BN-Lx and SHR left ventricle total RNA RNA-Seq
========================================================

* 4 samples per strain (SHR1, SHR2, SHR3, SHR4, BNLx1, BNLx2, BNLx3, BNLx4)
* ribosomal RNA depleted total RNA
* 2X100 reads using the stranded protocol
* sequencing done by Beckman Coulter Genomics
* synthetic spike-in were added to each sample (???)
  * Information regarding the spike-ins can be found here: http://products.invitrogen.com/ivgn/product/4456739

1. Unzip Files
---------------
```
qsub -q smp /home/saba/BNLx.SHR.Heart.totalRNA/programs/unzip.yucca.sh
```

2. Determine number of reads sent for each sample
-------------------------------------------------
```
qsub -q smp /home/saba/BNLx.SHR.Heart.totalRNA/programs/rawReadCounts.yucca.sh
```

```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BNLx.SHR/RNA-Seq.Liver/totalRNA.UCD/")
rawCounts = read.table(file="data/rawReadCounts.07Apr14.txt",sep=" ",header=FALSE,fill=TRUE)
rawCounts = rawCounts[is.na(rawCounts$V3),]
rawCounts$readFrag = as.numeric(rawCounts$V4)/4
rawCounts$strain = NA
rawCounts$strain[grep("BNLX_",rawCounts$V5)] = "BNLx"
rawCounts$strain[grep("SHR_",rawCounts$V5)] = "SHR"
rawCounts$sample = paste(rawCounts$strain,unlist(lapply(strsplit(rawCounts$V5,split="_",fixed=TRUE),function(a) a[2])),sep="")
rawCounts$lane = unlist(lapply(strsplit(rawCounts$V5,split="_",fixed=TRUE),function(a) a[4]))

readFragments = aggregate(rawCounts$readFrag,by=list(sample=rawCounts$sample,strain=rawCounts$strain,lane=rawCounts$lane),sum)
readFragments$numPairedReads = prettyNum(readFragments$x/2,big.mark=",",scientific=FALSE)
readFragments$numReadFragments = prettyNum(readFragments$x,big.mark=",",scientific=FALSE)

readFragments=readFragments[,colnames(readFragments)!="x"]
```

Raw Reads/Read Fragments From UCD Genomic Core
---------------------------

```{r, results='asis',echo=FALSE}
kable(readFragments,align=rep("c",ncol(readFragments)))
```
