---
title: "RI Panel - Batch 4 (Brain) Small RNA RNA-Seq"
author: "Laura Saba"
date: "July 1, 2015"
output:
  md_document:
    variant: markdown_github
---
RI Panel - Batch 4 (Brain) Small RNA RNA-Seq
===================

* SHR1, SHR3, and BNLx3
* tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic
* small RNA (<200 bp)
* 1X50 reads using the unstranded protocol
* sequencing done by UCD Genomics Core - received May 14, 2015
  * labeled 150508_7001413_0258_AC6K0DANXX  

Processing Done on PhenoGen Server  

1. Unzip Files - DONE
---------------
```
gzip -d /hi-seq/RI.Brain.smallRNA.14May15/rawReads/*.gz
```

2. Determine number of reads sent for each sample - DONE
-------------------------------------------------
```
/hi-seq/RI.Brain.smallRNA.14May15/programs/countRawReads.brain.smRNA.batch4.sh &
```
```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("~/Documents/NextGenSeq/Rat.RI.Brain.smallRNA.batch4/")
library(knitr)
rawCounts = read.table(file="data/rawReadCounts.30Jun15.txt",sep=" ",header=FALSE,fill=TRUE)

rawCounts$reads = rawCounts$V3
rawCounts$file = rawCounts$V1
rawCounts$sample = unlist(lapply(strsplit(rawCounts$file,split="/",fixed=TRUE),function(x) x[length(x)]))
rawCounts$sample = unlist(lapply(strsplit(rawCounts$sample,split="_",fixed=TRUE),function(x) paste(x[1:2],collapse="_")))

forPrint = rawCounts[,c("sample","reads")]
forPrint$reads = prettyNum(forPrint$reads,big.mark=",",scientific=FALSE)
colnames(forPrint) = c("sample","Number of Reads")
```

Raw Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint,align=rep("c",ncol(forPrint)),row.names=FALSE)
```

Total Number of Reads:  `r prettyNum(sum(rawCounts$reads),big.mark=",",scientific=FALSE)`  
Average Number of Reads Per Sample: `r prettyNum(sum(rawCounts$reads)/nrow(rawCounts),big.mark=",",scientific=FALSE)`  








