<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>RI Panel - Batch 1 (Brain) Total RNA RNA-Seq</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>RI Panel - Batch 1 (Brain) Total RNA RNA-Seq</h1>

<ul>
<li>2 samples per strain (SHR, BXH12, HXB13, HXB17, HXB2, HXB25, HXB27, HXB7)</li>
<li>tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic</li>
<li>ribosomal RNA depleted total RNA</li>
<li>2X100 reads using the stranded protocol</li>
<li>sequencing done by Beckman Coulter Genomics - received September 5, 2014</li>
<li>synthetic spike-in were added to each sample 

<ul>
<li>SHR_1, BXH12_1, HXB13_1, HXB17_1, HXB2_1, HXB25_1, HXB27_1, and HXB7_1 have mix 1; SHR_2, BXH12_2, HXB13_2, HXB17_2, HXB2_2, HXB25_2, HXB27_2, and HXB7_2 have mix 2</li>
<li>4 mL of 1:100 diluted spikes were added after RNA extraction but before rRNA-depletion</li>
<li>Information regarding the spike-ins can be found here: <a href="http://products.invitrogen.com/ivgn/product/4456739">http://products.invitrogen.com/ivgn/product/4456739</a></li>
</ul></li>
</ul>

<h2>1. Unzip Files - DONE</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch1/programs/unzip.yucca.sh
</code></pre>

<h2>2. Determine number of reads sent for each sample - DONE</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch1/programs/countRawReads.brain.batch1.sh
</code></pre>

<h2>Raw Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BXH12_1</td>
<td align="center">158,661,335</td>
<td align="center">317,322,670</td>
</tr>
<tr>
<td align="center">BXH12_2</td>
<td align="center">189,172,520</td>
<td align="center">378,345,040</td>
</tr>
<tr>
<td align="center">HXB13_1</td>
<td align="center">137,084,983</td>
<td align="center">274,169,966</td>
</tr>
<tr>
<td align="center">HXB13_2</td>
<td align="center">132,515,930</td>
<td align="center">265,031,860</td>
</tr>
<tr>
<td align="center">HXB17_1</td>
<td align="center">132,884,089</td>
<td align="center">265,768,178</td>
</tr>
<tr>
<td align="center">HXB17_2</td>
<td align="center">206,344,183</td>
<td align="center">412,688,366</td>
</tr>
<tr>
<td align="center">HXB2_1</td>
<td align="center">105,150,313</td>
<td align="center">210,300,626</td>
</tr>
<tr>
<td align="center">HXB2_2</td>
<td align="center">130,036,249</td>
<td align="center">260,072,498</td>
</tr>
<tr>
<td align="center">HXB25_1</td>
<td align="center">115,639,769</td>
<td align="center">231,279,538</td>
</tr>
<tr>
<td align="center">HXB25_2</td>
<td align="center">178,617,581</td>
<td align="center">357,235,162</td>
</tr>
<tr>
<td align="center">HXB27_1</td>
<td align="center">120,993,425</td>
<td align="center">241,986,850</td>
</tr>
<tr>
<td align="center">HXB27_2</td>
<td align="center">145,364,335</td>
<td align="center">290,728,670</td>
</tr>
<tr>
<td align="center">HXB7_1</td>
<td align="center">128,794,212</td>
<td align="center">257,588,424</td>
</tr>
<tr>
<td align="center">HXB7_2</td>
<td align="center">195,878,926</td>
<td align="center">391,757,852</td>
</tr>
<tr>
<td align="center">SHR_1</td>
<td align="center">156,973,256</td>
<td align="center">313,946,512</td>
</tr>
<tr>
<td align="center">SHR_2</td>
<td align="center">169,578,776</td>
<td align="center">339,157,552</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads: 2,403,689,882<br/>
Total Number of Read Fragments:  4,807,379,764<br/>
Average Number of Paired End Reads Per Sample: 150,230,618  </p>

<h2>3. Trim reads for adaptors and for quality - RUNNING</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch1/programs/trimReads.brain.batch1.sh
</code></pre>

</body>

</html>

