RNA-Seq Analysis of C57 vs DBA from Sanger Institute 
========================================================

Data were downloaded from the European Nucleotide Archive (ENA).  Paired-end reads (76X2)

## Download Data
```
#two DBA samples
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR033/ERR033015/ERR033015_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR033/ERR033015/ERR033015_2.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR033/ERR033016/ERR033016_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR033/ERR033016/ERR033016_2.fastq.gz
#two C57 samples
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR033/ERR033006/ERR033006_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR033/ERR033006/ERR033006_2.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR033/ERR033006/ERR033007_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR033/ERR033006/ERR033007_2.fastq.gz
```

## Unzip files
```
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/unzip.yucca.sh
```

## Determine number of reads sent for each sample

```
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/rawReadCounts.sh
```




### Raw Reads/Read Fragments From Sanger Institute

| sample | strain | numPairedReads | numReadFragments |
|:------:|:------:|:--------------:|:----------------:|
| C571   |  C57   |   35,752,923   |    71,505,846    |
| C572   |  C57   |   40,940,770   |    81,881,540    |
| DBA1   |  DBA   |   25,912,031   |    51,824,062    |
| DBA2   |  DBA   |   31,275,311   |    62,550,622    |


## Trim Reads for Adaptors and for Quality 

```
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/trimReads.sh
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/trimReads.v2.sh
```

## Characterize Trimmed Reads 

```
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/countTrimmed.sh
```




### Effect of Trimming Adaptors and Poor Quality Base Calls

| sample | strain | num of paired end reads | num of read fragments | avg trimmed read fragment length | num of read fragments after trimming | perc of read fragments eliminated during trimming |
|:------:|:------:|:-----------------------:|:---------------------:|:--------------------------------:|:------------------------------------:|:-------------------------------------------------:|
|  C571  |  C57   |       35,752,923        |      71,505,846       |              72.89               |              69,014,508              |                      0.035%                       |
|  C572  |  C57   |       40,940,770        |      81,881,540       |              71.80               |              77,291,488              |                      0.056%                       |
|  DBA1  |  DBA   |       25,912,031        |      51,824,062       |              74.01               |              51,018,388              |                      0.016%                       |
|  DBA2  |  DBA   |       31,275,311        |      62,550,622       |              73.73               |              60,763,532              |                      0.029%                       |



## Create Bowtie2 Index without spikes

```
cd /data/Tabastore3/LauraS/index
cat /data/Tabastore3/LauraS/BXD/RNA-Seq/GeneNetwork/index/mm10.revised.wo.chrY.fa /data/Tabastore3/LauraS/BXD/RNA-Seq/GeneNetwork/index/mm10.revised.chrY.v2.fa > mm10.cleaned.v1.fa
awk '{if($1~">") print $1; else print $0}' mm10.cleaned.v1.fa > mm10.cleaned.fa
```

```
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/createBowtie2Index.sh
```

#### Code from createBowtie2Index.sh
```
#!/bin/bash
module add bio/bowtie2
bowtie2-build /home/saba/index/mm10.cleaned.fa /home/saba/index/mm10.cleaned
```

## Alignment of Raw Reads to Reference Genome

```
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/tophatAlign.sh
```

#### Code from tophatAlign.sh
```
#!/bin/bash
module add bio/bowtie2
module add bio/tophat
tophat2 -o /home/saba/C57.DBA.Brain.Sanger/alignedReads/C571 -p 16 /home/saba/index/mm10.cleaned /home/saba/C57.DBA.Brain.Sanger/trimmedReads/ERR033006_1_val_1.fq /home/saba/C57.DBA.Brain.Sanger/trimmedReads/ERR033006_2_val_2.fq
tophat2 -o /home/saba/C57.DBA.Brain.Sanger/alignedReads/C572 -p 16 /home/saba/index/mm10.cleaned /home/saba/C57.DBA.Brain.Sanger/trimmedReads/ERR033007_1_val_1.fq /home/saba/C57.DBA.Brain.Sanger/trimmedReads/ERR033007_2_val_2.fq
tophat2 -o /home/saba/C57.DBA.Brain.Sanger/alignedReads/DBA1 -p 16 /home/saba/index/mm10.cleaned /home/saba/C57.DBA.Brain.Sanger/trimmedReads/ERR033015_1_val_1.fq /home/saba/C57.DBA.Brain.Sanger/trimmedReads/ERR033015_2_val_2.fq
tophat2 -o /home/saba/C57.DBA.Brain.Sanger/alignedReads/DBA2 -p 16 /home/saba/index/mm10.cleaned /home/saba/C57.DBA.Brain.Sanger/trimmedReads/ERR033016_1_val_1.fq /home/saba/C57.DBA.Brain.Sanger/trimmedReads/ERR033016_2_val_2.fq
```

## Characterization of Aligned Reads



|                           Label                           |    C571    | C571.pct |    C572    | C572.pct |    DBA1    | DBA1.pct |    DBA2    | DBA2.pct |
|:---------------------------------------------------------:|:----------:|:--------:|:----------:|:--------:|:----------:|:--------:|:----------:|:--------:|
|                number of paired-end reads                 | 34,507,254 | 100.00%  | 38,645,744 | 100.00%  | 25,509,194 | 100.00%  | 30,381,766 | 100.00%  |
|           number of left reads mapped to genome           | 28,515,644 |  82.64%  | 31,512,205 |  81.54%  | 22,800,619 |  89.38%  | 27,131,493 |  89.30%  |
|          number of right reads mapped to genome           | 27,878,050 |  80.79%  | 30,498,465 |  78.92%  | 22,574,292 |  88.49%  | 26,907,822 |  88.57%  |
|   number of mapped left reads with multiple alignments    |  2,871,541 |  8.32%   |  3,211,704 |  8.31%   |  4,077,575 |  15.98%  |  4,858,006 |  15.99%  |
|   number of mapped right reads with multiple alignments   |  2,822,002 |  8.18%   |  3,127,193 |  8.09%   |  4,028,573 |  15.79%  |  4,819,571 |  15.86%  |
| number of mapped left reads with more than 20 alignments  |    297,390 |  0.86%   |    327,869 |  0.85%   |    226,284 |  0.89%   |    335,718 |  1.10%   |
| number of mapped right reads with more than 20 alignments |    297,395 |  0.86%   |    327,877 |  0.85%   |    226,282 |  0.89%   |    335,720 |  1.11%   |
|                number of mapped read pairs                | 23,922,052 |  69.32%  | 25,746,958 |  66.62%  | 21,192,965 |  83.08%  | 25,238,269 |  83.07%  |
|   number of mapped read pairs with multiple alignments    |  2,562,374 |  7.43%   |  2,825,503 |  7.31%   |  3,732,475 |  14.63%  |  4,489,957 |  14.78%  |
|  number of mapped read pairs with discordant alignments   |  4,061,389 |  11.77%  |  4,863,369 |  12.58%  |  2,810,656 |  11.02%  |  3,516,264 |  11.57%  |


## Sort and Merge Aligned Reads By Strain 
```
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/sortAndMerge.sh
```

## Strain-Specific Transcriptome Reconstruction
```
qsub -q smp-q /home/saba/C57.DBA.Brain.Sanger/programs/reconstruction.sh
```

## Merge Strain-Specific Transcriptomes

## Quantify Individual Samples

## Assess differential expression
