Compare Transcriptome-Guided Reconstructions Between C57 and DBA
========================================================

1.  Import GTF files and limit transcripts to those that have FPKM>1 in at least on strain and are larger than 150 bp (not including introns)
--------------------------

```{r, eval=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BXD/RNA-Seq/Sanger")

C57 <- read.table(file="reconstruction/C57.ensemblGuided/transcripts.gtf",sep="\t",header=FALSE)
C57$transcript_id = unlist(lapply(strsplit(C57$V9,split="; "),function(a) gsub("transcript_id ","",a[grep("transcript_id",a)])))
C57$gene_id = unlist(lapply(strsplit(C57$V9,split="; "),function(a) gsub("gene_id ","",a[grep("gene_id",a)])))
C57$fpkm = as.numeric(unlist(lapply(strsplit(C57$V9,split="; "),function(a) gsub("FPKM ","",a[grep("FPKM",a)]))))

C57.exons = C57[C57$V3=="exon",]
C57.exons$length = C57.exons$V5 - C57.exons$V4 + 1

tmp = cbind(aggregate(C57.exons$length,by=list(C57.exons$transcript_id),sum),aggregate(C57.exons$transcript_id,by=list(C57.exons$transcript_id),length)$x)
colnames(tmp) = c("transcript_id","length","numExons")


C57.transcripts = C57[C57$V3=="transcript",]
C57.transcripts = merge(C57.transcripts, tmp,by="transcript_id")
C57.transcripts = C57.transcripts[C57.transcripts$fpkm!=0 & C57.transcripts$length>300,]

## DBA ##
DBA <- read.table(file="reconstruction/DBA.ensemblGuided/transcripts.gtf",sep="\t",header=FALSE)
DBA$transcript_id = unlist(lapply(strsplit(DBA$V9,split="; "),function(a) gsub("transcript_id ","",a[grep("transcript_id",a)])))
DBA$gene_id = unlist(lapply(strsplit(DBA$V9,split="; "),function(a) gsub("gene_id ","",a[grep("gene_id",a)])))
DBA$fpkm = as.numeric(unlist(lapply(strsplit(DBA$V9,split="; "),function(a) gsub("FPKM ","",a[grep("FPKM",a)]))))

DBA.exons = DBA[DBA$V3=="exon",]
DBA.exons$length = DBA.exons$V5 - DBA.exons$V4 + 1

tmp = cbind(aggregate(DBA.exons$length,by=list(DBA.exons$transcript_id),sum),aggregate(DBA.exons$transcript_id,by=list(DBA.exons$transcript_id),length)$x)
colnames(tmp) = c("transcript_id","length","numExons")


DBA.transcripts = DBA[DBA$V3=="transcript",]
DBA.transcripts = merge(DBA.transcripts, tmp,by="transcript_id")
DBA.transcripts = DBA.transcripts[DBA.transcripts$fpkm!=0 & DBA.transcripts$length>300,]
save(C57.transcripts,DBA.transcripts,C57,DBA,file="Rdata/origRecon.Rdata")
```

```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BXD/RNA-Seq/Sanger")
load("Rdata/origRecon.Rdata")
```

Comparison of Transcriptomes
----------------------------

* Number of Genes  
DBA = `r length(unique(DBA.transcripts$gene_id))`  
ISS = `r length(unique(ISS.transcripts$gene_id))`  

* Number of Transcripts  
ILS = `r nrow(ILS.transcripts)`  
ISS = `r nrow(ISS.transcripts)`  

* Average Number of Transcripts Per Gene  
ILS = `r nrow(ILS.transcripts)/length(unique(ILS.transcripts$gene_id))`  
ISS = `r nrow(ISS.transcripts)/length(unique(ISS.transcripts$gene_id))`  

* Max Number of Transcripts Per Gene  
ILS = `r max(table(ILS.transcripts$gene_id))`  
ISS = `r max(table(ISS.transcripts$gene_id))`  

* Number of RefSeq Genes Recovered    
ILS = `r sum(grepl("N",ILS.transcripts$transcript_id))`  
ISS = `r sum(grepl("N",ISS.transcripts$transcript_id))`  


2. Concatenate novel transcripts between strains, merge expressed RefSeq transcripts, and created BED File
--------------------------------------------------------------------------
```{r,eval=FALSE}

DBA.transcripts = DBA.transcripts[,c("transcript_id","gene_id","fpkm","length","numExons","V1","V4","V5")]
colnames(DBA.transcripts) = paste("DBA.",colnames(DBA.transcripts),sep="")

C57.transcripts = C57.transcripts[,c("transcript_id","gene_id","fpkm","length","numExons","V1","V4","V5")]
colnames(C57.transcripts) = paste("C57.",colnames(C57.transcripts),sep="")

DBA.transcripts$transcript_id = gsub("CUFF","DBA",DBA.transcripts$DBA.transcript_id)
C57.transcripts$transcript_id = gsub("CUFF","C57",C57.transcripts$C57.transcript_id)

transcripts = merge(DBA.transcripts[DBA.transcripts$DBA.fpkm>1,],C57.transcripts[C57.transcripts$C57.fpkm>1,],by="transcript_id",all=TRUE)

##  Convert to BED File  ##

C57.v2 = C57
C57.v2$transcript_id = gsub("CUFF","C57",C57.v2$transcript_id)
C57.v2$gene_id = gsub("CUFF","C57",C57.v2$gene_id)

DBA.v2 = DBA
DBA.v2$transcript_id = gsub("CUFF","DBA",DBA.v2$transcript_id)
DBA.v2$gene_id = gsub("CUFF","DBA",DBA.v2$gene_id)

combinedGTF = rbind(C57.v2,DBA.v2)
combinedGTF = combinedGTF[,!(colnames(combinedGTF) %in% c("V9","fpkm"))]
combinedGTF = combinedGTF[!duplicated(combinedGTF),]
combinedGTF = combinedGTF[combinedGTF$transcript_id %in% transcripts$transcript_id,]

gtf <- combinedGTF[combinedGTF$V3=="exon",]
gtf <- gtf[order(gtf$transcript_id,gtf$V4,gtf$V5),]

print(paste("Number of Genes ",length(unique(gtf$gene_id)),sep=""))
print(paste("Number of Transcripts ",length(unique(gtf$transcript_id)),sep=""))

##  create one per transcript ##
start <- aggregate(gtf$V4,by=list(gtf$transcript_id),function(a) paste(a,collapse=","))
stop <- aggregate(gtf$V5,by=list(gtf$transcript_id),function(a) paste(a,collapse=","))
transcripts <- gtf[!duplicated(gtf[,c("transcript_id","V1")]),c("transcript_id","gene_id","V1","V6","V7")]
onePerTrans <- merge(start,stop,by=1)
onePerTrans <- merge(transcripts,onePerTrans,by=1)

onePerTrans$exonNum = unlist(lapply(strsplit(onePerTrans$x.x,split=",",fixed=TRUE),length))
onePerTrans$start <- as.integer(unlist(lapply(strsplit(onePerTrans$x.x,split=",",fixed=TRUE),function(a) min(as.numeric(a)))))
onePerTrans$stop <- as.integer(unlist(lapply(strsplit(onePerTrans$x.y,split=",",fixed=TRUE),function(a) max(as.numeric(a)))))

## export as BED style file for overlap  ##
write.table(onePerTrans[,c("V1","start","stop","transcript_id","V6","V7")],file="data/combined.05May14.v1.bed",sep="\t",quote=FALSE,row.names=FALSE,col.names=FALSE)
```


3.  Look for overlap between transcripts within the combined BED file  
-----------------------------------------
BEDtools version = bedtools v2.16.2-zip-5db7ff9


```
export PATH=/usr/local/bedtools2/bin:$PATH
 
#cd /data/Tabastore3/LauraS/BXD/RNA-Seq/Sanger/data
cd /Volumes/LauraS/BXD/RNA-Seq/Sanger/data
intersectBed -a combined.05May14.v1.bed -b combined.05May14.v1.bed -wo > overlap.ensGuided.txt
```

4.  Identify novel transcripts that were identified in both BN-Lx and SHR 
------------------------------------------------------
Two transcripts were "merged" into one transcripts if:  
* they are both assigned to the same strand or one/both did not have a strand designation
* they were identified as novel in opposite strains (one from SHR and one from BNLx)
* if 1) all exon starts and exon stops matched OR 2) all exon junctions matched, transcription start and stop sites could differ OR 3) two one-exon transcripts with transcription start sites within 100 bp of each other and transcription stop sites within 100 bp of each other

```{r,eval=FALSE}
overlap <- read.table(file="data/overlap.ensGuided.txt",sep="\t",header=FALSE)
colnames(overlap) <- c(paste(c("chr","start","stop","transcript_id","V5","strand"),rep(c("A","B"),each=6),sep="."),"overlap")

## add exon information to overlaps
setA = setB = onePerTrans[,c("transcript_id","x.x","x.y","exonNum")]
colnames(setA) = c("transcript_id.A","exonStarts.A","exonStops.A","exonNum.A")
colnames(setB) = c("transcript_id.B","exonStarts.B","exonStops.B","exonNum.B")

overlap = merge(overlap,setA,by="transcript_id.A")
overlap = merge(overlap,setB,by="transcript_id.B")

overlap$ID = c(1:nrow(overlap))

##  identify exon junction

findJunct <- function(starts,stops){
  start <- strsplit(starts,split=",",fixed=TRUE)[[1]]
  stop <- strsplit(stops,split=",",fixed=TRUE)[[1]]
  junct <- NA
  if(length(start)>1) junct <- paste(paste(stop[-length(stop)],start[-1],sep="//"),collapse=",")
  return(junct)
  }
  
overlap$exonJunct.A = apply(overlap[,c("exonStarts.A","exonStops.A")],1,function(a) findJunct(a[1],a[2]))
overlap$exonJunct.B = apply(overlap[,c("exonStarts.B","exonStops.B")],1,function(a) findJunct(a[1],a[2]))



## remove overlap that is on the opposite strand
overlap <- overlap[overlap$strand.A=="." | overlap$strand.B=="." | overlap$strand.A==overlap$strand.B,]

## remove overlap between same transcript
overlap <- overlap[overlap$transcript_id.A!=overlap$transcript_id.B,]


## remove overlap where coding regions don't overlap  ##
intronCheck =c()
for(i in 1:nrow(overlap)){
  input = overlap[i,]
  out = sum(sum(unlist(lapply(strsplit(strsplit(input[1,"exonJunct.B"],split=",")[[1]],split="//"),function(a) input[1,"start.A"]>a[1] & input[1,"stop.A"]<a[2]))),sum(unlist(lapply(strsplit(strsplit(input[1,"exonJunct.A"],split=",")[[1]],split="//"),function(a) input[1,"start.B"]>a[1] & input[1,"stop.B"]<a[2]))),na.rm=TRUE)
  intronCheck = c(intronCheck,out)
  }
  
overlap = overlap[intronCheck==0,]

##  remove overlap with annotated genes/transcripts  ##
overlap = overlap[!(grepl("ENS",overlap$transcript_id.B) | grepl("ENS",overlap$transcript_id.A)),]

##  remove within strain overlap ##
overlap = overlap[!(grepl("C57",overlap$transcript_id.B) & grepl("C57",overlap$transcript_id.A)),]
overlap = overlap[!(grepl("DBA",overlap$transcript_id.B) & grepl("DBA",overlap$transcript_id.A)),]

## transcripts without any overlap - 47,361 transcripts
noOverlap = onePerTrans[!(onePerTrans$transcript_id %in% unique(c(overlap$transcript_id.A,overlap$transcript_id.B))),]

##  perfect overlap - 32 transcripts
perfect = overlap[overlap$exonStarts.A==overlap$exonStarts.B & overlap$exonStops.A==overlap$exonStops.B,]

##  identify exon junction matches - 1,740 transcripts
reduced = overlap[!(overlap$ID %in% perfect$ID),]
exonJunctMatch <- reduced[!is.na(reduced$exonJunct.A) & !is.na(reduced$exonJunct.B) & reduced$exonJunct.A==reduced$exonJunct.B,]

##  examine one-exon transcripts - 8026 transcripts
reduced = overlap[!(overlap$ID %in% c(perfect$ID,exonJunctMatch$ID)),]
oneExons = reduced[reduced$exonNum.A==1 & reduced$exonNum.B==1,]

closeMatch.oneExon = oneExons[abs(oneExons$start.A - oneExons$start.B)<100 & abs(oneExons$stop.A - oneExons$stop.B)<100, ]
sum(duplicated(closeMatch.oneExon$transcript_id.B))
sum(duplicated(closeMatch.oneExon$transcript_id.A))

##  Check for overlap between renamed  ##
matched = as.data.frame(rbind(perfect,exonJunctMatch,closeMatch.oneExon))
sum(duplicated(matched$transcript_id.B))

##  create new transcript for matched transcripts - 8,105 matches
matched = matched[grep("C57",matched$transcript_id.A),]

matched$transcript_id = paste("total",c(1:nrow(matched)),sep=".")
matched$gene_id = NA
matched$V1 = matched$chr.A
matched$V6 = 1000
matched$V7 = matched$strand.A
matched$start = apply(matched[,c("start.B","start.A")],1,min)
matched$stop = apply(matched[,c("stop.B","stop.A")],1,max)
matched$exonNum = matched$exonNum.A

matched$x.x = NA
matched$x.x[matched$start==matched$start.A] = matched$exonStarts.A[matched$start==matched$start.A]
matched$x.x[matched$start==matched$start.B] = matched$exonStarts.B[matched$start==matched$start.B]

matched$x.y = NA
matched$x.y[matched$stop==matched$stop.A] = matched$exonStops.A[matched$stop==matched$stop.A]
matched$x.y[matched$stop==matched$stop.B] = matched$exonStops.B[matched$stop==matched$stop.B]

##  BAND-AID - double match  ##
matched = matched[matched$transcript_id.A!="C57.13198.2",]

matched.toAdd = matched[,colnames(onePerTrans)]

new.transcripts = onePerTrans[!(onePerTrans$transcript_id %in% c(matched$transcript_id.A,matched$transcript_id.B)),]
new.transcripts = rbind(new.transcripts,matched.toAdd)

write.table(new.transcripts[,c("V1","start","stop","transcript_id","V6","V7")],file="data/newTotal.05May14.v2.bed",sep="\t",quote=FALSE,row.names=FALSE,col.names=FALSE)
```

5.  Identify overlap between transcripts in order to identify transcripts from the same gene
------------------------------------------------------

BEDtools Version = bedtools v2.16.2-zip-5db7ff9
```
cd /Volumes/LauraS/BXD/RNA-Seq/Sanger/data
intersectBed -a newTotal.05May14.v2.bed -b newTotal.05May14.v2.bed -wo > overlap.total.forGeneID.txt
```

6.  Match transcripts to genes and create new GTF file  
------------------------------------------------------

Two transcripts are from the same gene if:  
* Their transcription start sites matched exactly OR
* Their transcription stop sites matched exactly OR
* At least one exon-exon junction matched exactly

```{r,eval=FALSE}

findJunct <- function(starts,stops){
  start <- strsplit(starts,split=",",fixed=TRUE)[[1]]
  stop <- strsplit(stops,split=",",fixed=TRUE)[[1]]
  junct <- NA
  if(length(start)>1) junct <- paste(paste(stop[-length(stop)],start[-1],sep="//"),collapse=",")
  return(junct)
  }
  
new.transcripts$exonJunct = apply(new.transcripts[,c("x.x","x.y")],1,function(a) findJunct(a[1],a[2]))

overlap = read.table(file="data/overlap.total.forGeneID.txt",sep="\t",header=FALSE)
overlap = overlap[overlap$V4!=overlap$V10,]
overlap = overlap[overlap$V6=="." | overlap$V12=="." | overlap$V6==overlap$V12,]

overlap$pair = apply(overlap,1,function(a) paste(sort(a[c("V4","V10")]),collapse="//"))
overlap = overlap[!duplicated(overlap$pair),]

pairs = c()
for(i in 1:nrow(overlap)){
  x = new.transcripts[new.transcripts$transcript_id==overlap$V4[i],]
  y = new.transcripts[new.transcripts$transcript_id==overlap$V10[i],]
  sameGene = FALSE
  if(x$start==y$start | x$stop==y$stop) sameGene = TRUE
  if(!is.na(y$exonJunct) & !is.na(x$exonJunct) & sum(duplicated(c(unlist(strsplit(x$exonJunct,split=",")),unlist(strsplit(y$exonJunct,split=",")))))>0) sameGene = TRUE
  if(sameGene) pairs = rbind(pairs,c(x$transcript_id,y$transcript_id))
}


##  identify genes with multiple transcripts and create gene identifier  ##
library(igraph)
library("ggm")

edges <- pairs
graph.polyA <- graph.data.frame(edges)
graph.polyA <- igraph.to.graphNEL(graph.polyA)

compList <- connectedComp(graph.polyA)
names(compList) <- paste("total",c(1:length(compList)),sep=".")

multiTrans <- sapply(names(compList),function(a) cbind(gene_id.new=a,transcript_id.new=paste(a,c(1:length(compList[[a]])),sep="."),id=compList[[a]]))
multiTrans <- data.frame(do.call(rbind,multiTrans))

singleTrans <- data.frame(id=new.transcripts$transcript_id[!(new.transcripts$transcript_id %in% multiTrans$id)])
singleTrans$gene_id.new = paste("total",c((length(compList)+1):(length(compList)+nrow(singleTrans))),sep=".")
singleTrans$transcript_id.new = paste(singleTrans$gene_id.new,1,sep=".")

updated <- merge(new.transcripts,rbind(multiTrans,singleTrans),by.x="transcript_id",by.y="id")

##  Create New GTF File  ##

byExon = data.frame(do.call(rbind,apply(updated,1,function(tmp) data.frame(transcript_id.new = as.character(tmp[13]),V4=unlist(strsplit(as.character(tmp[6]),split=",")),V5=unlist(strsplit(as.character(tmp[7]),split=","))))))

gtf = merge(updated,byExon,by="transcript_id.new")
#gtf$V1 = paste("chr",gtf$V1,sep="")
gtf$V2 = "CuffLinks"
gtf$V3 = "exon"
gtf$V8 = "."
gtf$V9 = paste('gene_id "',gtf$gene_id.new,'"; transcript_id "',gtf$transcript_id.new,'"; original "',gtf$transcript_id,'";',sep="")

gtf = gtf[,paste("V",c(1:9),sep="")]

write.table(gtf,file="reconstruction/transcript.total.05May14.gtf",row.names=FALSE,col.names=FALSE,quote=FALSE,sep="\t")
```

7. Use SailFish To Check Quantification In New GTF
---------------------------------------

### Extract sequence of transcripts
```
# index genome
cd /Volumes/LauraS/index
samtools faidx mm10.cleaned.fa

# generate fasta file of transcript sequences
cd /Volumes/LauraS/BXD/RNA-Seq/Sanger/reconstruction
gffread -w transcript.total.05May14.fa -g /Volumes/LauraS/index/mm10.cleaned.fa transcript.total.05May14.gtf
```

### Index Transcript Sequences

```
export LD_LIBRARY_PATH=/usr/local/sailfish/lib:$LD_LIBRARY_PATH
export PATH=/usr/local/sailfish/bin:$PATH
cd /data/Tabastore3/LauraS/BXD/RNA-Seq/Sanger/reconstruction/
sailfish index -t transcript.total.05May14.fa -k 20 -p 5 -o C57.DBA.brain.transcriptome.v1

cd /data/Tabastore3/LauraS/BXD/RNA-Seq/Sanger/
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v1 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033006_1_val_1.fq -2 ./trimmedReads/ERR033006_2_val_2.fq -p 4 -o ./quantification/sailFish.v1/C571
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v1 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033007_1_val_1.fq -2 ./trimmedReads/ERR033007_2_val_2.fq -p 4 -o ./quantification/sailFish.v1/C572

sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v1 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033015_1_val_1.fq -2 ./trimmedReads/ERR033015_2_val_2.fq -p 4 -o ./quantification/sailFish.v1/DBA1
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v1 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033016_1_val_1.fq -2 ./trimmedReads/ERR033016_2_val_2.fq -p 4 -o ./quantification/sailFish.v1/DBA2
```


8.  After re-quantifying with new GTF, eliminate transcripts with less than 1 FPKM in both strains and rename transcripts based on their combined (across strains) FPKM value, i.e., the transcript label '.1' is the major transcript
--------------------------------------------------

```{r,eval=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BXD/RNA-Seq/Sanger")

C571.count = read.table(file="quantification/sailFish.v1/C571/quant_bias_corrected.sf",sep="\t",header=FALSE)
C572.count = read.table(file="quantification/sailFish.v1/C572/quant_bias_corrected.sf",sep="\t",header=FALSE)
DBA1.count = read.table(file="quantification/sailFish.v1/DBA1/quant_bias_corrected.sf",sep="\t",header=FALSE)
DBA2.count = read.table(file="quantification/sailFish.v1/DBA2/quant_bias_corrected.sf",sep="\t",header=FALSE)

colnames(C571.count) = c("transcript","length",paste("C571",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(C572.count) = c("transcript","length",paste("C572",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(DBA1.count) = c("transcript","length",paste("DBA1",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(DBA2.count) = c("transcript","length",paste("DBA2",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))

counts = merge(C571.count,C572.count,by=c("transcript","length"))
counts = merge(counts,DBA1.count,by=c("transcript","length"))
counts = merge(counts,DBA2.count,by=c("transcript","length"))

counts$numSamplesGrt1 = rowSums(counts[,grep("TPM",colnames(counts))]>1)
counts = counts[counts$numSamplesGrt1!=0,]
counts$gene_id = paste("total",unlist(lapply(strsplit(counts$transcript,split=".",fixed=TRUE),function(a) a[2])),sep=".")

counts$sumTPM = rowSums(counts[,grep("TPM",colnames(counts))])
counts = counts[order(counts$gene_id,counts$sumTPM,decreasing=TRUE),]

## rename transcripts based on TPM  ##
newID = data.frame(do.call(rbind,aggregate(counts$transcript,by=list(gene_id=counts$gene_id),function(a) cbind(a,paste("total",strsplit(a,split=".",fixed=TRUE)[[1]][2],1:length(a),sep=".")))$x))
colnames(newID) = c("transcript_id","transcript_id.new")

##  reduce GTF file  ##
orig.gtf = read.table(file="reconstruction/transcript.total.05May14.gtf",header=FALSE,sep="\t")
orig.gtf$transcript_id = unlist(lapply(strsplit(orig.gtf$V9,split="; "),function(a) gsub("transcript_id ","",a[grep("transcript",a)])))
orig.gtf$gene_id = unlist(lapply(strsplit(orig.gtf$V9,split="; "),function(a) gsub("gene_id ","",a[grep("gene",a)])))

new.gtf = merge(orig.gtf,newID,by="transcript_id")
new.gtf$V9 = paste('gene_id "',new.gtf$gene_id,'"; transcript_id "',new.gtf$transcript_id.new,'";',sep="")
write.table(new.gtf[,paste("V",c(1:9),sep="")],file="reconstruction/transcript.total.07May14.v2.gtf",row.names=FALSE,col.names=FALSE,quote=FALSE,sep="\t")
```

9.  Re-quantify again based on reduced GTF
----------------------------------

```
cd /Volumes/LauraS/BXD/RNA-Seq/Sanger/reconstruction
gffread -w transcript.total.07May14.v2.fa -g /Volumes/LauraS/index/mm10.cleaned.fa transcript.total.07May14.v2.gtf

export LD_LIBRARY_PATH=/usr/local/sailfish/lib:$LD_LIBRARY_PATH
export PATH=/usr/local/sailfish/bin:$PATH
cd /data/Tabastore3/LauraS/BXD/RNA-Seq/Sanger/reconstruction/
sailfish index -t transcript.total.07May14.v2.fa -k 20 -p 5 -o C57.DBA.brain.transcriptome.v2

cd /data/Tabastore3/LauraS/BXD/RNA-Seq/Sanger/
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v2 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033006_1_val_1.fq -2 ./trimmedReads/ERR033006_2_val_2.fq -p 4 -o ./quantification/sailFish.v2/C571
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v2 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033007_1_val_1.fq -2 ./trimmedReads/ERR033007_2_val_2.fq -p 4 -o ./quantification/sailFish.v2/C572

sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v2 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033015_1_val_1.fq -2 ./trimmedReads/ERR033015_2_val_2.fq -p 4 -o ./quantification/sailFish.v2/DBA1
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v2 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033016_1_val_1.fq -2 ./trimmedReads/ERR033016_2_val_2.fq -p 4 -o ./quantification/sailFish.v2/DBA2

```

```{r,eval=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BXD/RNA-Seq/Sanger")

C571.count = read.table(file="quantification/sailFish.v2/C571/quant_bias_corrected.sf",sep="\t",header=FALSE)
C572.count = read.table(file="quantification/sailFish.v2/C572/quant_bias_corrected.sf",sep="\t",header=FALSE)
DBA1.count = read.table(file="quantification/sailFish.v2/DBA1/quant_bias_corrected.sf",sep="\t",header=FALSE)
DBA2.count = read.table(file="quantification/sailFish.v2/DBA2/quant_bias_corrected.sf",sep="\t",header=FALSE)

colnames(C571.count) = c("transcript","length",paste("C571",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(C572.count) = c("transcript","length",paste("C572",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(DBA1.count) = c("transcript","length",paste("DBA1",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(DBA2.count) = c("transcript","length",paste("DBA2",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))

counts = merge(C571.count,C572.count,by=c("transcript","length"))
counts = merge(counts,DBA1.count,by=c("transcript","length"))
counts = merge(counts,DBA2.count,by=c("transcript","length"))

counts$numSamplesGrt1 = rowSums(counts[,grep("TPM",colnames(counts))]>1)
counts = counts[counts$numSamplesGrt1!=0,]
counts$gene_id = paste("total",unlist(lapply(strsplit(counts$transcript,split=".",fixed=TRUE),function(a) a[2])),sep=".")

counts$sumTPM = rowSums(counts[,grep("TPM",colnames(counts))])
counts = counts[order(counts$gene_id,counts$sumTPM,decreasing=TRUE),]

## rename transcripts based on TPM  ##
newID = data.frame(do.call(rbind,aggregate(counts$transcript,by=list(gene_id=counts$gene_id),function(a) cbind(a,paste("total",strsplit(a,split=".",fixed=TRUE)[[1]][2],1:length(a),sep=".")))$x))
colnames(newID) = c("transcript_id","transcript_id.new")

##  reduce GTF file  ##
orig.gtf = read.table(file="reconstruction/transcript.total.07May14.v2.gtf",header=FALSE,sep="\t")
orig.gtf$transcript_id = unlist(lapply(strsplit(orig.gtf$V9,split="; "),function(a) gsub("transcript_id ","",gsub(";","",a[grep("transcript",a)]))))
orig.gtf$gene_id = unlist(lapply(strsplit(orig.gtf$V9,split="; "),function(a) gsub("gene_id ","",a[grep("gene",a)])))

new.gtf = merge(orig.gtf,newID,by="transcript_id")
new.gtf$V9 = paste('gene_id "',new.gtf$gene_id,'"; transcript_id "',new.gtf$transcript_id.new,'";',sep="")
write.table(new.gtf[,paste("V",c(1:9),sep="")],file="reconstruction/transcript.total.08May14.v3.gtf",row.names=FALSE,col.names=FALSE,quote=FALSE,sep="\t")
```



10.  Re-quantify again based on reduced GTF
----------------------------------

```
cd /Volumes/LauraS/BXD/RNA-Seq/Sanger/reconstruction
gffread -w transcript.total.08May14.v3.fa -g /Volumes/LauraS/index/mm10.cleaned.fa transcript.total.08May14.v3.gtf

export LD_LIBRARY_PATH=/usr/local/sailfish/lib:$LD_LIBRARY_PATH
export PATH=/usr/local/sailfish/bin:$PATH
cd /data/Tabastore3/LauraS/BXD/RNA-Seq/Sanger/reconstruction/
sailfish index -t transcript.total.08May14.v3.fa -k 20 -p 5 -o C57.DBA.brain.transcriptome.v3

cd /data/Tabastore3/LauraS/BXD/RNA-Seq/Sanger/
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v3 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033006_1_val_1.fq -2 ./trimmedReads/ERR033006_2_val_2.fq -p 4 -o ./quantification/sailFish.v3/C571
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v3 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033007_1_val_1.fq -2 ./trimmedReads/ERR033007_2_val_2.fq -p 4 -o ./quantification/sailFish.v3/C572

sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v3 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033015_1_val_1.fq -2 ./trimmedReads/ERR033015_2_val_2.fq -p 4 -o ./quantification/sailFish.v3/DBA1
sailfish quant -i ./reconstruction/C57.DBA.brain.transcriptome.v3 -l "T=PE:O=><:S=U" -1 ./trimmedReads/ERR033016_1_val_1.fq -2 ./trimmedReads/ERR033016_2_val_2.fq -p 4 -o ./quantification/sailFish.v3/DBA2

```


```{r,eval=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BXD/RNA-Seq/Sanger")

C571.count = read.table(file="quantification/sailFish.v3/C571/quant_bias_corrected.sf",sep="\t",header=FALSE)
C572.count = read.table(file="quantification/sailFish.v3/C572/quant_bias_corrected.sf",sep="\t",header=FALSE)
DBA1.count = read.table(file="quantification/sailFish.v3/DBA1/quant_bias_corrected.sf",sep="\t",header=FALSE)
DBA2.count = read.table(file="quantification/sailFish.v3/DBA2/quant_bias_corrected.sf",sep="\t",header=FALSE)

colnames(C571.count) = c("transcript","length",paste("C571",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(C572.count) = c("transcript","length",paste("C572",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(DBA1.count) = c("transcript","length",paste("DBA1",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))
colnames(DBA2.count) = c("transcript","length",paste("DBA2",c("TPM","RPKM","KPKM","EstNumKmers","EstNumReads"),sep="."))

counts = merge(C571.count,C572.count,by=c("transcript","length"))
counts = merge(counts,DBA1.count,by=c("transcript","length"))
counts = merge(counts,DBA2.count,by=c("transcript","length"))

counts$numSamplesGrt1 = rowSums(counts[,grep("TPM",colnames(counts))]>1)
counts = counts[counts$numSamplesGrt1!=0,]
counts$gene_id = paste("total",unlist(lapply(strsplit(counts$transcript,split=".",fixed=TRUE),function(a) a[2])),sep=".")

counts$sumTPM = rowSums(counts[,grep("TPM",colnames(counts))])
counts = counts[order(counts$gene_id,counts$sumTPM,decreasing=TRUE),]

## rename transcripts based on TPM  ##
newID = data.frame(do.call(rbind,aggregate(counts$transcript,by=list(gene_id=counts$gene_id),function(a) cbind(a,paste("total",strsplit(a,split=".",fixed=TRUE)[[1]][2],1:length(a),sep=".")))$x))
colnames(newID) = c("transcript_id","transcript_id.new")

##  reduce GTF file  ##
orig.gtf = read.table(file="reconstruction/transcript.total.08May14.v3.gtf",header=FALSE,sep="\t")
orig.gtf$transcript_id = unlist(lapply(strsplit(orig.gtf$V9,split="; "),function(a) gsub("transcript_id ","",gsub(";","",a[grep("transcript",a)]))))
orig.gtf$gene_id = unlist(lapply(strsplit(orig.gtf$V9,split="; "),function(a) gsub("gene_id ","",a[grep("gene",a)])))

new.gtf = merge(orig.gtf,newID,by="transcript_id")
new.gtf$V9 = paste('gene_id "',new.gtf$gene_id,'"; transcript_id "',new.gtf$transcript_id.new,'";',sep="")
write.table(new.gtf[,paste("V",c(1:9),sep="")],file="reconstruction/transcript.total.08May14.v4.gtf",row.names=FALSE,col.names=FALSE,quote=FALSE,sep="\t")
```


###  Rename Files to Represent Final Transcriptome
cd /Volumes/LauraS/BXD/RNA-Seq/Sanger/reconstruction
mv transcript.total.08May14.v3.gtf transcript.total.08May14.FINAL.gtf
mv transcript.total.08May14.v3.fa transcript.total.08May14.FINAL.fa

###  Create GTF file that can be uploaded to UCSC
```{r}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BXD/RNA-Seq/Sanger")

gtf = read.table(file="reconstruction/transcript.total.08May14.FINAL.gtf",sep="\t",header=FALSE)
gtf$V1 = paste("chr",gtf$V1,sep="")
write.table(gtf,file="reconstruction/transcript.total.08May14.FINAL.ucsc.gtf",sep="\t",row.names=FALSE,col.names=FALSE,quote=FALSE)


gtf$transcript_id = unlist(lapply(strsplit(gtf$V9,split=";",fixed=TRUE),function(a) gsub(" transcript_id ","",a[2])))
gtf$gene_id = unlist(lapply(strsplit(gtf$V9,split=";",fixed=TRUE),function(a) gsub("gene_id ","",a[1])))
```



###   Make BedGraph Files To Display on Browser
```
module load bio/bedtools2
bedtools genomecov -bg -ibam /home/saba/C57.DBA.Brain.Sanger/alignedReads/C57.brain.bam > /home/saba/C57.DBA.Brain.Sanger/alignedReads/C57.brain.bedgraph
bedtools genomecov -bg -ibam /home/saba/C57.DBA.Brain.Sanger/alignedReads/DBA.brain.bam > /home/saba/C57.DBA.Brain.Sanger/alignedReads/DBA.brain.bedgraph



