---
title: "BNLx and SHR Females - Liver Small RNA RNA-Seq"
author: "Laura Saba"
date: "July 1, 2015"
output:
  md_document:
    variant: markdown_github
---

BNLx and SHR Females - Liver Small RNA RNA-Seq
========================================================

* 4 samples per strain (BN-Lx and SHR)
* tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic
* small RNA (<200 bp)
* 1X50 reads using the unstranded protocol
* sequencing done by UCD Genomics Core - received April 9, 2015
* labeled 150403_7001413_0234_AC6R43ANXX

1. Unzip Files - DONE
---------------
```
gzip -d /hi-seq/BNLx.SHR.Liver.Females.smallRNA/rawReads/*.gz & 
```

2. Determine number of reads sent for each sample - 
-------------------------------------------------
```
/hi-seq/BNLx.SHR.Liver.Females.smallRNA/programs/countRawReads.liver.females.smRNA.sh &
```
```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
library(knitr)
setwd("~/Documents/NextGenSeq/Rat.RI.Liver.smallRNA.batch2/")
rawCounts = read.table(file="data/rawReadCounts.01Jul15.txt",sep=" ",header=FALSE,fill=TRUE)

rawCounts$reads = rawCounts$V3
rawCounts$file = rawCounts$V1
rawCounts$sample = unlist(lapply(strsplit(rawCounts$file,split="/",fixed=TRUE),function(x) x[length(x)]))
rawCounts$sample = unlist(lapply(strsplit(rawCounts$sample,split="_",fixed=TRUE),function(x) gsub("LIVER","",paste(x[1:2],collapse="_"))))

rawCounts$sample[rawCounts$sample=="HXB_1"]="HXB1_1"
rawCounts$sample[rawCounts$sample=="XXB15_1"]="HXB15_1"

forPrint = rawCounts[,c("sample","reads")]
forPrint$reads = prettyNum(forPrint$reads,big.mark=",",scientific=FALSE)
colnames(forPrint) = c("sample","Number of Reads")
```

Raw Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint,align=rep("c",ncol(forPrint)),row.names=FALSE)
```

Total Number of Reads:  `r prettyNum(sum(rawCounts$reads),big.mark=",",scientific=FALSE)`  
Average Number of Reads Per Sample: `r prettyNum(sum(rawCounts$reads)/nrow(rawCounts),big.mark=",",scientific=FALSE)`  

