<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>BNLx and SHR Brain Transcriptome Reconstruction</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>BNLx and SHR Brain Transcriptome Reconstruction</h1>

<h2>rRNA-Depleted Total RNA Reconstruction</h2>

<p>Complete SHR Transcriptome:<br/>
52,086 genes<br/>
58,015 transcripts<br/>
5,458 Ensembl transcripts<br/>
52,557 novel transcripts  </p>

<p>Complete BNLx Transcriptome:<br/>
215,955 genes<br/>
236,143 transcripts<br/>
14,411 Ensembl transcripts<br/>
221,732 novel transcripts  </p>

<h2>1. Limit to transcripts with greater than 50 coverage and a transcript length greater than 200 nt (not including introns)</h2>

<p>High Confidence SHR Transcripts:<br/>
7,916 genes<br/>
9,460 transcripts<br/>
3,497 Ensembl transcripts<br/>
1,892 novel transcripts of Ensembl genes<br/>
4,071 novel transcripts/genes   </p>

<p>High Confidence BNLx Transcripts:<br/>
28,020 genes<br/>
32,094 transcripts<br/>
8,441 Ensembl transcripts<br/>
5,104 novel transcripts of Ensembl genes<br/>
18,549 novel transcripts/genes   </p>

<h2>2. Identify overlap between novel transcripts in strain-specific reconstructions</h2>

<p>BEDtools version = bedtools v2.19.1</p>

<pre><code>export PATH=/usr/local/bedtools2/bin:$PATH

cd /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Brain.total/reconstruction/tmp
intersectBed -a shr.bed -b bnlx.bed -wo &gt; overlap.brain.total.byStrain.txt
</code></pre>

<h2>3.  Compare novel transcripts across strain-specific transcriptomes BN-Lx and SHR </h2>

<p>Two transcripts were &ldquo;merged&rdquo; into one transcripts if:  </p>

<ul>
<li>they are both assigned to the same strand or one/both did not have a strand designation</li>
<li>they were identified as novel in opposite strains (one from SHR and one from BNLx)</li>
<li>if 1) all exon starts and exon stops matched OR 2) all exon junctions matched, transcription start and stop sites could differ OR 3) two one-exon transcripts with transcription start sites within 100 bp of each other and transcription stop sites within 100 bp of each other</li>
</ul>

<p>Combined Transcriptome:  </p>

<ul>
<li>37,699 transcripts</li>
<li>1,162 novel transcripts identified in both strains</li>
<li>22,515 novel transcripts only identified in BNLx</li>
<li>4,822 novel transcripts only identified in SHR</li>
<li>9,177 Ensembl transcripts</li>
<li>23 novel transcripts very similar to Ensembl</li>
</ul>

<h2>4.  Identify overlap between transcripts in order to identify transcripts from the same gene</h2>

<p>BEDtools Version = bedtools v2.19.1</p>

<pre><code>export PATH=/usr/local/bedtools2/bin:$PATH
cd /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Brain.total/reconstruction/tmp
intersectBed -a combined.brain.total.22Jul14.bed -b combined.brain.total.22Jul14.bed -wo &gt; overlap.brain.total.forGeneID.txt
</code></pre>

<h2>5.  Match transcripts to genes and create new GTF file  </h2>

<p>Two transcripts are from the same gene if:  </p>

<ul>
<li>Their transcription start sites matched exactly OR</li>
<li>Their transcription stop sites matched exactly OR</li>
<li>At least one exon-exon junction matched exactly</li>
</ul>

<h2>6. Quantitate Combined Transcriptome (1st Iteration)</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/quant.Recon.22Jul14.v1.SHR.sh
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/quant.Recon.22Jul14.v1.BNLx.sh
</code></pre>

<pre><code>## [1] 0.9214
</code></pre>

<p>In the first iteration, 34,737 out of 37,699 transcripts were retained (30,964 out of 33,077 genes).</p>

<h2>7. Quantitate Combined Transcriptome (2nd Iteration)</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/quant.Recon.23Jul14.v2.SHR.sh
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/quant.Recon.23Jul14.v2.BNLx.sh
</code></pre>

<pre><code>## [1] 0.9987
</code></pre>

<p>In the second iteration, 34,692 out of 34,737 transcripts were retained (30,936 out of 30,964 genes).</p>

</body>

</html>

