<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>RI Panel - Batch 3 (Brain) Total RNA RNA-Seq</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>RI Panel - Batch 3 (Brain) Total RNA RNA-Seq</h1>

<ul>
<li>1 sample per strain (SHR, BXH13, BXH5, BXH8)</li>
<li>2 samples per strain (BNLx, BXH10, BXH11, BXH3, BXH6, BXH9)</li>
<li>tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic</li>
<li>ribosomal RNA depleted total RNA</li>
<li>2X100 reads using the stranded protocol</li>
<li>sequencing done by Beckman Coulter Genomics - received November 22, 2014

<ul>
<li>labeled UColorado Tabakoff Batch 5 by Beckman</li>
</ul></li>
<li>synthetic spike-in were added to each sample 

<ul>
<li>BXH3_1, BXH3_2, BXH5_1, BXH8_1, BXH10_1, BXH11_2, BXH13_1, and SHR_1 have mix 1</li>
<li>BXH6_1, BXH6_2, BXH9_1, BXH9_2, BXH10_2, BXH11_1, BNLx_1, and BNLx_2 have mix 2</li>
<li>4 mL of 1:100 diluted spikes were added after RNA extraction but before rRNA-depletion</li>
<li>Information regarding the spike-ins can be found here: <a href="http://products.invitrogen.com/ivgn/product/4456739">http://products.invitrogen.com/ivgn/product/4456739</a></li>
</ul></li>
</ul>

<h2>1. Unzip Files - DONE</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch3/programs/unzip.yucca.sh
</code></pre>

<h2>2. Determine number of reads sent for each sample - DONE</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch3/programs/countRawReads.brain.batch3.sh
</code></pre>

<h2>Raw Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLX_1</td>
<td align="center">149,664,346</td>
<td align="center">299,328,692</td>
</tr>
<tr>
<td align="center">BNLX_2</td>
<td align="center">147,573,455</td>
<td align="center">295,146,910</td>
</tr>
<tr>
<td align="center">BXH10_1</td>
<td align="center">136,119,802</td>
<td align="center">272,239,604</td>
</tr>
<tr>
<td align="center">BXH10_2</td>
<td align="center">143,559,349</td>
<td align="center">287,118,698</td>
</tr>
<tr>
<td align="center">BXH11_1</td>
<td align="center">143,165,227</td>
<td align="center">286,330,454</td>
</tr>
<tr>
<td align="center">BXH11_2</td>
<td align="center">142,142,287</td>
<td align="center">284,284,574</td>
</tr>
<tr>
<td align="center">BXH13_1</td>
<td align="center">136,469,855</td>
<td align="center">272,939,710</td>
</tr>
<tr>
<td align="center">BXH3_1</td>
<td align="center">148,297,982</td>
<td align="center">296,595,964</td>
</tr>
<tr>
<td align="center">BXH3_2</td>
<td align="center">138,276,728</td>
<td align="center">276,553,456</td>
</tr>
<tr>
<td align="center">BXH5_1</td>
<td align="center">141,525,963</td>
<td align="center">283,051,926</td>
</tr>
<tr>
<td align="center">BXH6_1</td>
<td align="center">140,776,240</td>
<td align="center">281,552,480</td>
</tr>
<tr>
<td align="center">BXH6_2</td>
<td align="center">146,263,134</td>
<td align="center">292,526,268</td>
</tr>
<tr>
<td align="center">BXH8_1</td>
<td align="center">130,847,183</td>
<td align="center">261,694,366</td>
</tr>
<tr>
<td align="center">BXH9_1</td>
<td align="center">132,797,003</td>
<td align="center">265,594,006</td>
</tr>
<tr>
<td align="center">BXH9_2</td>
<td align="center">133,839,175</td>
<td align="center">267,678,350</td>
</tr>
<tr>
<td align="center">SHR_1</td>
<td align="center">146,707,331</td>
<td align="center">293,414,662</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads: 2,258,025,060<br/>
Total Number of Read Fragments:  4,516,050,120<br/>
Average Number of Paired End Reads Per Sample: 141,126,566  </p>

<h2>3. Trim reads for adaptors and for quality - RUNNING (job number 44238)</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch3/programs/trimReads.brain.batch3.sh
</code></pre>

</body>

</html>

