<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>BNLx and SHR Liver rRNA-Depleted Total RNA From UCD Genomics Core</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>BNLx and SHR Liver rRNA-Depleted Total RNA From UCD Genomics Core</h1>

<ul>
<li>6 liver ribosomal RNA depleted total RNA samples, SHR1, SHR5, SHR25, BNLx1, BNLx2, and BNLx3</li>
<li>synthetic spikes included Mix 1 in SHR1, BNLx1, and BNLx2 and Mix 2 in SHR5, SHR25, and BNLx3</li>
<li>2X100 paired end reads using the stranded protocol</li>
<li>received 4/1/14</li>
</ul>

<h2>1. Unzip FASTQ files on Yucca - DONE</h2>

<pre><code>qsub -q smp-q /home/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/unzip.yucca.sh
</code></pre>

<h2>2. Determine number of reads sent for each sample - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/countRawReads.liver.sh
</code></pre>

<h2>Raw Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">119,624,256</td>
<td align="center">239,248,512</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">104,944,610</td>
<td align="center">209,889,220</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">69,001,559</td>
<td align="center">138,003,118</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">109,987,327</td>
<td align="center">219,974,654</td>
</tr>
<tr>
<td align="center">SHR25</td>
<td align="center">82,591,985</td>
<td align="center">165,183,970</td>
</tr>
<tr>
<td align="center">SHR5</td>
<td align="center">96,770,957</td>
<td align="center">193,541,914</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads: 582,920,694<br/>
Total Number of Read Fragments:  1,165,841,388<br/>
Average Number of Paired End Reads Per Sample: 97,153,449  </p>

<h2>3. Trim Reads for Adaptors and for Quality - DONE</h2>

<pre><code>qsub -q smp-q /home/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/trimReads.yucca.sh
</code></pre>

<h2>4. Characterizing Trimmed Reads - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/countTrimmedReads.liver.sh
</code></pre>

<h2>Trimmed Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
<th align="center">Average Read Fragment Length After Trimming (first read fragment)</th>
<th align="center">Average Read Fragment Length After Trimming (second read fragment)</th>
<th align="center">Number of Read Fragments After Trimming</th>
<th align="center">Percent of Read Fragments That Remained After Trimming</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">119,624,256</td>
<td align="center">239,248,512</td>
<td align="center">94.2</td>
<td align="center">91.4</td>
<td align="center">234,327,370</td>
<td align="center">97.9%</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">104,944,610</td>
<td align="center">209,889,220</td>
<td align="center">96.6</td>
<td align="center">95.6</td>
<td align="center">207,427,684</td>
<td align="center">98.8%</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">69,001,559</td>
<td align="center">138,003,118</td>
<td align="center">84.7</td>
<td align="center">97.2</td>
<td align="center">137,036,336</td>
<td align="center">99.3%</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">109,987,327</td>
<td align="center">219,974,654</td>
<td align="center">97.8</td>
<td align="center">96.7</td>
<td align="center">217,663,066</td>
<td align="center">98.9%</td>
</tr>
<tr>
<td align="center">SHR25</td>
<td align="center">82,591,985</td>
<td align="center">165,183,970</td>
<td align="center">84.2</td>
<td align="center">96.9</td>
<td align="center">163,641,336</td>
<td align="center">99.1%</td>
</tr>
<tr>
<td align="center">SHR5</td>
<td align="center">96,770,957</td>
<td align="center">193,541,914</td>
<td align="center">96.5</td>
<td align="center">95.3</td>
<td align="center">190,584,770</td>
<td align="center">98.5%</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads After Trimming: 575,340,281<br/>
Total Number of Read Fragments After Trimming:  1,150,680,562<br/>
Average Number of Paired End Reads Per Sample After Trimming: 95,890,047  </p>

<h2>5.  Generate Strain-Specific Genomes Including Spike-In Sequences - DONE</h2>

<pre><code>qsub -q smp-q /home/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/createGenomeIndex.yucca.sh
</code></pre>

<h2>6. Align trimmed reads to ribosomal RNA - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/alignTo.rRNA.liver.01Jul14.sh
</code></pre>

<p><strong>move files</strong></p>

<pre><code>get BNLx1.rRNA/align_summary.txt rRNA.summary.BNLx1.txt  
get BNLx2.rRNA/align_summary.txt rRNA.summary.BNLx2.txt  
get BNLx3.rRNA/align_summary.txt rRNA.summary.BNLx3.txt  
get SHR1.rRNA/align_summary.txt rRNA.summary.SHR1.txt  
get SHR25.rRNA/align_summary.txt rRNA.summary.SHR25.txt  
get SHR5.rRNA/align_summary.txt rRNA.summary.SHR5.txt  
</code></pre>

<table><thead>
<tr>
<th align="center">Sample</th>
<th align="center">Number of Paired-End Reads After Trimming</th>
<th align="center">Number of Paired-End Reads with at Least One Fragment Aligned to rRNA</th>
<th align="center">Percent of Paired-End Reads Aligned to rRNA</th>
<th align="center">Number of Paired-End Reads NOT Aligned to rRNA</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">117,163,685</td>
<td align="center">27,582,382</td>
<td align="center">23.5%</td>
<td align="center">89,581,303</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">103,713,842</td>
<td align="center">22,154,258</td>
<td align="center">21.4%</td>
<td align="center">81,559,584</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">68,518,168</td>
<td align="center">4,224,768</td>
<td align="center">6.2%</td>
<td align="center">64,293,400</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">108,831,533</td>
<td align="center">6,658,641</td>
<td align="center">6.1%</td>
<td align="center">102,172,892</td>
</tr>
<tr>
<td align="center">SHR25</td>
<td align="center">81,820,668</td>
<td align="center">7,966,721</td>
<td align="center">9.7%</td>
<td align="center">73,853,947</td>
</tr>
<tr>
<td align="center">SHR5</td>
<td align="center">95,292,385</td>
<td align="center">6,749,851</td>
<td align="center">7.1%</td>
<td align="center">88,542,534</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads After Eliminating rRNA Aligned Reads: 500,003,660<br/>
Average Number of Paired End Reads Per Sample After Eliminating rRNA Aligned Reads: 83,333,943  </p>

<h2>7. Convert unmapped.bam into fastq files - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/convertToFastQ.liver.sh
</code></pre>

<h2>8. Alignment of Raw Reads to Strain-Specific Genomes With Mitochondrial Chromosome - RUNNING (7/7/14)</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/tophatAlign.liver.sh
</code></pre>

<p><strong><em>STOPPED HERE</em></strong></p>

<h2>9.  Characterization of Aligned Reads</h2>

<p><strong>move files</strong></p>

<pre><code>cd /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/alignedReads
get ./BNLx1.withMT/align_summary.txt align_summary.BNLx1.txt
get ./BNLx2.withMT/align_summary.txt align_summary.BNLx2.txt
get ./BNLx3.withMT/align_summary.txt align_summary.BNLx3.txt
get ./SHR1.withMT/align_summary.txt align_summary.SHR1.txt
get ./SHR5.withMT/align_summary.txt align_summary.SHR5.txt
get ./SHR25.withMT/align_summary.txt align_summary.SHR25.txt
</code></pre>

<h3>Statistics on Alignment to Genome</h3>

<h2>10.  Examine Spike-In Expression - HAVEN&#39;T RUN FOR DATA WITH MITO</h2>

<pre><code>qsub -q smp-q /home/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/quantifyControlSpikes.21Apr14.sh
</code></pre>

<h3>Synthetic spike-in summary</h3>

<h3>Comparison of raw read count versus spike concentrations</h3>

<p>Each color represents a different sample.  The y-axis of the plot on the left represents the log base 2 of the read count for each spike plus 1. The y-axis of the plot in the middle represents the &ldquo;voom&rdquo; expression values when the library size is calculated as ALL mapped reads. The y-axis of the plot on the right represents the &ldquo;voom&rdquo; expression values when the library size is calculated as only the reads that map to a synthetic spike-in.  </p>

<h3>Expected Differential Expression In Spikes Between Mixes</h3>

<p>Red lines represent the expected log 2 fold differences.  Only spikes that were included in a concentration greater than 0.5 in both samples were included in figure (66 out of 92 spikes).</p>

<h2>11.  Sort and Merge Aligned Reads By Strain</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/sortAndMerge.liver.sh
</code></pre>

<h2>12.  Strain-Specific Transcriptome Reconstruction (Ensembl-Guided)</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/reconSHR.liver.sh
qsub -q smp /home/data/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/reconBNLx.liver.sh
</code></pre>

<h2>13.  Create BigWig Files</h2>

<pre><code>qsub -q smp /home/saba/BNLx.SHR.Liver.totalRNA.UCD/programs/testBigWig.sh
</code></pre>

</body>

</html>

