<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>BN-Lx and SHR Female Brain Total RNA RNA-Seq</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>BN-Lx and SHR Female Brain Total RNA RNA-Seq</h1>

<ul>
<li>4 samples per strain (SHR1, SHR2, SHR3, SHR4, BNLx1, BNLx2, BNLx3, BNLx4)</li>
<li>ribosomal RNA depleted total RNA</li>
<li>2X100 reads using the stranded protocol</li>
<li>sequencing done by Beckman Coulter Genomics - received June 18th, 2014

<ul>
<li>sequenced on HiSeq2500 (rapid runs)</li>
<li>each sample was pooled with a sample from the opposite strain and each pool was put into two lanes of the HiSeq2500</li>
</ul></li>
<li>synthetic spike-in were added to each sample 

<ul>
<li>SHR1, SHR3, BNLx2, and BNLx4 have mix 1; SHR2, SHR4, BNLx1, and BNLx3 have mix 2</li>
<li>4 microliters of 1:100 diluted spikes were add after RNA extraction but before rRNA-depletion</li>
<li>Information regarding the spike-ins can be found here: <a href="http://products.invitrogen.com/ivgn/product/4456739">http://products.invitrogen.com/ivgn/product/4456739</a></li>
</ul></li>
</ul>

<h2>1. Unzip Files </h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.Females.totalRNA/programs/unzip.yucca.sh
</code></pre>

<h2>2. Determine number of reads sent for each sample </h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.Females.totalRNA/programs/countRawReads.femaleBrain.sh
</code></pre>

<h2>Raw Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">lane</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">L001</td>
<td align="center">62,784,996</td>
<td align="center">125,569,992</td>
</tr>
<tr>
<td align="center">BNLx1</td>
<td align="center">L002</td>
<td align="center">64,290,267</td>
<td align="center">128,580,534</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">L001</td>
<td align="center">68,391,044</td>
<td align="center">136,782,088</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">L002</td>
<td align="center">68,269,835</td>
<td align="center">136,539,670</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">L001</td>
<td align="center">64,358,304</td>
<td align="center">128,716,608</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">L002</td>
<td align="center">66,204,000</td>
<td align="center">132,408,000</td>
</tr>
<tr>
<td align="center">BNLx4</td>
<td align="center">L001</td>
<td align="center">51,229,872</td>
<td align="center">102,459,744</td>
</tr>
<tr>
<td align="center">BNLx4</td>
<td align="center">L002</td>
<td align="center">51,698,100</td>
<td align="center">103,396,200</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">L001</td>
<td align="center">60,154,945</td>
<td align="center">120,309,890</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">L002</td>
<td align="center">61,642,412</td>
<td align="center">123,284,824</td>
</tr>
<tr>
<td align="center">SHR2</td>
<td align="center">L001</td>
<td align="center">49,555,245</td>
<td align="center">99,110,490</td>
</tr>
<tr>
<td align="center">SHR2</td>
<td align="center">L002</td>
<td align="center">49,452,858</td>
<td align="center">98,905,716</td>
</tr>
<tr>
<td align="center">SHR3</td>
<td align="center">L001</td>
<td align="center">62,786,750</td>
<td align="center">125,573,500</td>
</tr>
<tr>
<td align="center">SHR3</td>
<td align="center">L002</td>
<td align="center">64,265,863</td>
<td align="center">128,531,726</td>
</tr>
<tr>
<td align="center">SHR4</td>
<td align="center">L001</td>
<td align="center">68,375,841</td>
<td align="center">136,751,682</td>
</tr>
<tr>
<td align="center">SHR4</td>
<td align="center">L002</td>
<td align="center">68,643,223</td>
<td align="center">137,286,446</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads: 982,103,555<br/>
Total Number of Read Fragments:  1,964,207,110<br/>
Average Number of Paired End Reads Per Sample: 122,762,944  </p>

<h2>3. Trim reads for adaptors and for quality</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.Females.totalRNA/programs/trimReads.femaleBrain.sh
</code></pre>

<h2>4. Characterize Trimmed Reads</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.Females.totalRNA/programs/countTrimmedReads.femaleBrain.sh
</code></pre>

<h2>Trimmed Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">lane</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
<th align="center">Average Read Fragment Length After Trimming (first read fragment)</th>
<th align="center">Average Read Fragment Length After Trimming (second read fragment)</th>
<th align="center">Number of Read Fragments After Trimming</th>
<th align="center">Percent of Read Fragments That Remained After Trimming</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">L001</td>
<td align="center">62,784,996</td>
<td align="center">125,569,992</td>
<td align="center">97.8</td>
<td align="center">96.9</td>
<td align="center">123,837,806</td>
<td align="center">98.6%</td>
</tr>
<tr>
<td align="center">BNLx1</td>
<td align="center">L002</td>
<td align="center">64,290,267</td>
<td align="center">128,580,534</td>
<td align="center">97.8</td>
<td align="center">96.8</td>
<td align="center">126,788,580</td>
<td align="center">98.6%</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">L001</td>
<td align="center">68,391,044</td>
<td align="center">136,782,088</td>
<td align="center">96.6</td>
<td align="center">95.4</td>
<td align="center">135,013,118</td>
<td align="center">98.7%</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">L002</td>
<td align="center">68,269,835</td>
<td align="center">136,539,670</td>
<td align="center">96.5</td>
<td align="center">95.3</td>
<td align="center">134,735,488</td>
<td align="center">98.7%</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">L001</td>
<td align="center">64,358,304</td>
<td align="center">128,716,608</td>
<td align="center">98.1</td>
<td align="center">96.9</td>
<td align="center">127,437,742</td>
<td align="center">99.0%</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">L002</td>
<td align="center">66,204,000</td>
<td align="center">132,408,000</td>
<td align="center">98.1</td>
<td align="center">96.9</td>
<td align="center">131,065,632</td>
<td align="center">99.0%</td>
</tr>
<tr>
<td align="center">BNLx4</td>
<td align="center">L001</td>
<td align="center">51,229,872</td>
<td align="center">102,459,744</td>
<td align="center">96.8</td>
<td align="center">96.0</td>
<td align="center">101,490,762</td>
<td align="center">99.1%</td>
</tr>
<tr>
<td align="center">BNLx4</td>
<td align="center">L002</td>
<td align="center">51,698,100</td>
<td align="center">103,396,200</td>
<td align="center">96.8</td>
<td align="center">96.0</td>
<td align="center">102,416,740</td>
<td align="center">99.1%</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">L001</td>
<td align="center">60,154,945</td>
<td align="center">120,309,890</td>
<td align="center">98.6</td>
<td align="center">97.3</td>
<td align="center">118,772,152</td>
<td align="center">98.7%</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">L002</td>
<td align="center">61,642,412</td>
<td align="center">123,284,824</td>
<td align="center">98.6</td>
<td align="center">97.2</td>
<td align="center">121,686,888</td>
<td align="center">98.7%</td>
</tr>
<tr>
<td align="center">SHR2</td>
<td align="center">L001</td>
<td align="center">49,555,245</td>
<td align="center">99,110,490</td>
<td align="center">96.4</td>
<td align="center">95.1</td>
<td align="center">98,184,392</td>
<td align="center">99.1%</td>
</tr>
<tr>
<td align="center">SHR2</td>
<td align="center">L002</td>
<td align="center">49,452,858</td>
<td align="center">98,905,716</td>
<td align="center">96.3</td>
<td align="center">95.0</td>
<td align="center">97,976,326</td>
<td align="center">99.1%</td>
</tr>
<tr>
<td align="center">SHR3</td>
<td align="center">L001</td>
<td align="center">62,786,750</td>
<td align="center">125,573,500</td>
<td align="center">97.7</td>
<td align="center">96.4</td>
<td align="center">123,716,320</td>
<td align="center">98.5%</td>
</tr>
<tr>
<td align="center">SHR3</td>
<td align="center">L002</td>
<td align="center">64,265,863</td>
<td align="center">128,531,726</td>
<td align="center">97.7</td>
<td align="center">96.4</td>
<td align="center">126,610,316</td>
<td align="center">98.5%</td>
</tr>
<tr>
<td align="center">SHR4</td>
<td align="center">L001</td>
<td align="center">68,375,841</td>
<td align="center">136,751,682</td>
<td align="center">98.4</td>
<td align="center">97.2</td>
<td align="center">135,148,896</td>
<td align="center">98.8%</td>
</tr>
<tr>
<td align="center">SHR4</td>
<td align="center">L002</td>
<td align="center">68,643,223</td>
<td align="center">137,286,446</td>
<td align="center">98.4</td>
<td align="center">97.2</td>
<td align="center">135,665,804</td>
<td align="center">98.8%</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads After Trimming: 970,273,481<br/>
Total Number of Read Fragments After Trimming:  1,940,546,962<br/>
Average Number of Paired End Reads Per Sample After Trimming: 121,284,185  </p>

<h2>5. Align to ribosomal RNA </h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.Females.totalRNA/programs/alignTo.rRNA.femaleBrain.sh
</code></pre>

<p><strong>move files</strong></p>

<pre><code>get SHR1.rRNA/align_summary.txt rRNA.summary.SHR1.txt
get SHR2.rRNA/align_summary.txt rRNA.summary.SHR2.txt
get SHR3.rRNA/align_summary.txt rRNA.summary.SHR3.txt
get SHR4.rRNA/align_summary.txt rRNA.summary.SHR4.txt
get BNLx1.rRNA/align_summary.txt rRNA.summary.BNLx1.txt
get BNLx2.rRNA/align_summary.txt rRNA.summary.BNLx2.txt
get BNLx3.rRNA/align_summary.txt rRNA.summary.BNLx3.txt
get BNLx4.rRNA/align_summary.txt rRNA.summary.BNLx4.txt
</code></pre>

<h2>Proportion of Reads Aligned to rRNA</h2>

<table><thead>
<tr>
<th align="center">Sample</th>
<th align="center">Number of Paired-End Reads After Trimming</th>
<th align="center">Number of Paired-End Reads with at Least One Fragment Aligned to rRNA</th>
<th align="center">Percent of Paired-End Reads Aligned to rRNA</th>
<th align="center">Number of Paired-End Reads NOT Aligned to rRNA</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">125,313,193</td>
<td align="center">6,050,278</td>
<td align="center">4.8%</td>
<td align="center">119,262,915</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">134,874,303</td>
<td align="center">15,043,918</td>
<td align="center">11.2%</td>
<td align="center">119,830,385</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">129,251,687</td>
<td align="center">16,080,441</td>
<td align="center">12.4%</td>
<td align="center">113,171,246</td>
</tr>
<tr>
<td align="center">BNLx4</td>
<td align="center">101,953,751</td>
<td align="center">4,891,544</td>
<td align="center">4.8%</td>
<td align="center">97,062,207</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">120,229,520</td>
<td align="center">3,224,820</td>
<td align="center">2.7%</td>
<td align="center">117,004,700</td>
</tr>
<tr>
<td align="center">SHR2</td>
<td align="center">98,080,359</td>
<td align="center">10,168,864</td>
<td align="center">10.4%</td>
<td align="center">87,911,495</td>
</tr>
<tr>
<td align="center">SHR3</td>
<td align="center">125,163,318</td>
<td align="center">2,556,359</td>
<td align="center">2.0%</td>
<td align="center">122,606,959</td>
</tr>
<tr>
<td align="center">SHR4</td>
<td align="center">135,407,350</td>
<td align="center">3,074,718</td>
<td align="center">2.3%</td>
<td align="center">132,332,632</td>
</tr>
</tbody></table>

<p><strong>Total Number of Paired End Reads After Eliminating rRNA Aligned Reads:</strong> 909,182,539<br/>
<strong>Average Number of Paired End Reads Per Sample After Eliminating rRNA Aligned Reads:</strong> 113,647,817<br/>
<strong>Average Percent of Paired End Reads Eliminated Due to Alignment to rRNA:</strong> 6.0%  </p>

<h2>6.  Convert unmapped.bam file into fastq files - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.Females.totalRNA/programs/convertToFastQ.female.brain.sh
</code></pre>

<h2>7. Align to strain-specific genome - RUNNING (started 7/18/14)</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.Females.totalRNA/programs/tophatAlign.female.brain.sh
</code></pre>

<p><strong><em>STOPPED HERE</em></strong></p>

<h2>8. Sort and merge BAM files by strain</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Heart.totalRNA/programs/sortAndMerge.heart.sh
</code></pre>

<h2>9. Create BigWig Files</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Heart.totalRNA/programs/createBigWig.heart.sh
</code></pre>

<p>UCSC Browser Tracks
track type=bigWig name=&ldquo;SHR.plus.heart&rdquo; description=&ldquo;SHR Total RNA Heart - plus strand&rdquo; bigDataUrl=http://ucsc:<a href="mailto:JU7etr5t@phenogen.ucdenver.edu">JU7etr5t@phenogen.ucdenver.edu</a>/ucsc/SHR.heart.genome.plus.bw</p>

<p>track type=bigWig name=&ldquo;SHR.minus.heart&rdquo; description=&ldquo;SHR Total RNA Heart - minus strand&rdquo; bigDataUrl=http://ucsc:<a href="mailto:JU7etr5t@phenogen.ucdenver.edu">JU7etr5t@phenogen.ucdenver.edu</a>/ucsc/SHR.heart.genome.minus.bw</p>

<p>track type=bigWig name=&ldquo;BNLx.plus.heart&rdquo; description=&ldquo;BNLx Total RNA Heart - plus strand&rdquo; bigDataUrl=http://ucsc:<a href="mailto:JU7etr5t@phenogen.ucdenver.edu">JU7etr5t@phenogen.ucdenver.edu</a>/ucsc/BNLx.heart.genome.plus.bw</p>

<p>track type=bigWig name=&ldquo;BNLx.minus.heart&rdquo; description=&ldquo;BNLx Total RNA Heart - minus strand&rdquo; bigDataUrl=http://ucsc:<a href="mailto:JU7etr5t@phenogen.ucdenver.edu">JU7etr5t@phenogen.ucdenver.edu</a>/ucsc/BNLx.heart.genome.minus.bw</p>

<h2>10. Characterize aligned reads</h2>

<p><strong>move files</strong></p>

<pre><code>cd /home/data/saba/BNLx.SHR.Heart.totalRNA/alignedReads
get ./BNLx1.withMT/align_summary.txt align_summary.BNLx1.txt
get ./BNLx2.withMT/align_summary.txt align_summary.BNLx2.txt
get ./BNLx3.withMT/align_summary.txt align_summary.BNLx3.txt
get ./BNLx4.withMT/align_summary.txt align_summary.BNLx4.txt
get ./SHR1.withMT/align_summary.txt align_summary.SHR1.txt
get ./SHR2.withMT/align_summary.txt align_summary.SHR2.txt
get ./SHR3.withMT/align_summary.txt align_summary.SHR3.txt
get ./SHR4.withMT/align_summary.txt align_summary.SHR4.txt
</code></pre>

<h3>Statistics on Alignment to Genome</h3>

<h3>Characteristics of Alignments</h3>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Heart.totalRNA/programs/countByChrom.heart.sh
</code></pre>

<h3>Synthetic spike-in summary</h3>

<h2>9. Transcriptome Reconstruction (Ensembl-Guided)</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Heart.totalRNA/programs/reconSHR.heart.sh
qsub -q smp /home/data/saba/BNLx.SHR.Heart.totalRNA/programs/reconBNLx.heart.sh
</code></pre>

</body>

</html>

