BNLx and SHR Brain Transcriptome Reconstruction
========================================================

PolyA+ Reconstruction
--------------------
```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BNLx.SHR")

#######################
##  local functions  ##
#######################

# format GTF files to include transcript ids, gene ids, and fpkm values
gtfFormat = function(data){
  data$transcript_id = unlist(lapply(strsplit(data$V9,split=";",fixed=TRUE),function(a) gsub(" transcript_id ","",a[grep("transcript",a)]))) 
  data$gene_id = unlist(lapply(strsplit(data$V9,split=";",fixed=TRUE),function(a) gsub("gene_id ","",a[grep("gene",a)]))) 
  data$fpkm = as.numeric(unlist(lapply(strsplit(data$V9,split=";",fixed=TRUE),function(a) gsub(" FPKM ","",a[grep("FPKM",a)]))))
  return(data)
}
```

```{r,echo=FALSE}
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BNLx.SHR")

shr = read.table(file = "RNA-Seq.Brain.polyA/reconstruction/SHR.Brain.PolyA.ensemblGuided/transcripts.gtf",sep="\t",header=FALSE)
shr = gtfFormat(shr)

bnlx = read.table(file = "RNA-Seq.Brain.polyA/reconstruction/BNLx.Brain.PolyA.ensemblGuided/transcripts.gtf",sep="\t",header=FALSE)
bnlx = gtfFormat(bnlx)

shr.transcripts = shr[shr$V3=="transcript",c("V1","V4","V5","V7","transcript_id","gene_id","fpkm")]
bnlx.transcripts = bnlx[bnlx$V3=="transcript",c("V1","V4","V5","V7","transcript_id","gene_id","fpkm")]

anno = read.table(file="/Volumes/LauraS/annotation/ensemblGenes.Rnor_5.0.08Jul14.txt",sep="\t",header=TRUE)
anno$Description = unlist(lapply(strsplit(anno$Description,split="[",fixed=TRUE),function(a) gsub("[[:space:]]*$","",a[1])))

shr.transcripts = merge(shr.transcripts,anno,by.x="transcript_id",by.y="EnsemblTranscriptID",all.x=TRUE)
bnlx.transcripts = merge(bnlx.transcripts,anno,by.x="transcript_id",by.y="EnsemblTranscriptID",all.x=TRUE)

# generate gene annotation labels
anno.gene = shr.transcripts[!is.na(shr.transcripts$EnsemblGeneID),]
anno.gene = anno.gene[!duplicated(anno.gene[,c("gene_id","EnsemblGeneID")]),]

ensGeneID = aggregate(anno.gene$EnsemblGeneID,by=list(gene_id=anno.gene$gene_id),paste,collapse="//")
geneSymbol = aggregate(anno.gene$GeneSymbol,by=list(gene_id=anno.gene$gene_id),paste,collapse="//")
descrip = aggregate(anno.gene$Description,by=list(gene_id=anno.gene$gene_id),paste,collapse="//")
biotype = aggregate(anno.gene$GeneBiotype,by=list(gene_id=anno.gene$gene_id),function(a) paste(sort(unique(a)),collapse="//"))

geneAnno = merge(ensGeneID,geneSymbol,by="gene_id")
geneAnno = merge(geneAnno,descrip,by="gene_id")
colnames(geneAnno) = c("gene_id","EnsemblGeneID","GeneSymbol","Description")
geneAnno = merge(geneAnno,biotype,by="gene_id")
colnames(geneAnno) = c("gene_id","EnsemblGeneID","GeneSymbol","Description","geneBiotype")

shr.transcripts = shr[shr$V3=="transcript",c("V1","V4","V5","transcript_id","gene_id","fpkm")]
shr.transcripts = merge(shr.transcripts,geneAnno,by="gene_id",all.x=TRUE)
shr.transcripts = shr.transcripts[shr.transcripts$fpkm>0,]

# generate gene annotation labels - BNLx
anno.gene = bnlx.transcripts[!is.na(bnlx.transcripts$EnsemblGeneID),]
anno.gene = anno.gene[!duplicated(anno.gene[,c("gene_id","EnsemblGeneID")]),]

ensGeneID = aggregate(anno.gene$EnsemblGeneID,by=list(gene_id=anno.gene$gene_id),paste,collapse="//")
geneSymbol = aggregate(anno.gene$GeneSymbol,by=list(gene_id=anno.gene$gene_id),paste,collapse="//")
descrip = aggregate(anno.gene$Description,by=list(gene_id=anno.gene$gene_id),paste,collapse="//")
biotype = aggregate(anno.gene$GeneBiotype,by=list(gene_id=anno.gene$gene_id),function(a) paste(sort(unique(a)),collapse="//"))

geneAnno = merge(ensGeneID,geneSymbol,by="gene_id")
geneAnno = merge(geneAnno,descrip,by="gene_id")
colnames(geneAnno) = c("gene_id","EnsemblGeneID","GeneSymbol","Description")
geneAnno = merge(geneAnno,biotype,by="gene_id")
colnames(geneAnno) = c("gene_id","EnsemblGeneID","GeneSymbol","Description","geneBiotype")

bnlx.transcripts = bnlx[bnlx$V3=="transcript",c("V1","V4","V5","transcript_id","gene_id","fpkm")]
bnlx.transcripts = merge(bnlx.transcripts,geneAnno,by="gene_id",all.x=TRUE)
bnlx.transcripts = bnlx.transcripts[bnlx.transcripts$fpkm>0,]

threshold=3

ens.bnlx = bnlx.transcripts$transcript_id[grepl("ENS",bnlx.transcripts$transcript_id) & bnlx.transcripts$fpkm>threshold]
ens.shr = shr.transcripts$transcript_id[grepl("ENS",shr.transcripts$transcript_id) & shr.transcripts$fpkm>threshold]

length(ens.bnlx)
length(ens.shr)
sum(ens.bnlx %in% ens.shr)

sum(ens.bnlx %in% shr.transcripts$transcript_id[grepl("ENS",shr.transcripts$transcript_id)] )
sum(ens.shr %in% bnlx.transcripts$transcript_id[grepl("ENS",bnlx.transcripts$transcript_id)] )


###  Look For Overlap of High Confidence Transcripts  ###
bnlx.high = bnlx.transcripts[bnlx.transcripts$fpkm>3,]
shr.high = shr.transcripts[shr.transcripts$fpkm>3,]

colnames(bnlx.high) =  c("gene_id.bnlx","V1","V4","V5","transcript_id","fpkm.bnlx","EnsemblGeneID.bnlx","GeneSymbol.bnlx","Description.bnlx","geneBiotype.bnlx")  
colnames(shr.high) =  c("gene_id.shr","V1","V4","V5","transcript_id","fpkm.shr","EnsemblGeneID.shr","GeneSymbol.shr","Description.shr","geneBiotype.shr")

ens.high = merge(bnlx.high[grep("ENS",bnlx.high$transcript_id),],shr.high[grep("ENS",shr.high$transcript_id),],by=c("transcript_id","V1","V4","V5"),all=TRUE)

plot(log2(ens.high$fpkm.bnlx),log2(ens.high$fpkm.shr))
abline(a=0,b=1,col="red")

sum(ens.high$fpkm.bnlx<ens.high$fpkm.shr,na.rm=TRUE)/sum(!is.na(ens.high$fpkm.bnlx) & !is.na(ens.high$fpkm.shr))

shr.tracking = read.table(file = "RNA-Seq.Brain.polyA/reconstruction/SHR.Brain.PolyA.ensemblGuided/isoforms.fpkm_tracking",sep="\t",header=TRUE)

shr.tracking = shr.tracking[order(shr.tracking$FPKM),]
shr.tracking = shr.tracking[shr.tracking$FPKM>0,]


bnlx.tracking = read.table(file = "RNA-Seq.Brain.polyA/reconstruction/BNLx.Brain.PolyA.ensemblGuided/isoforms.fpkm_tracking",sep="\t",header=TRUE)

bnlx.tracking = bnlx.tracking[order(bnlx.tracking$FPKM),]
bnlx.tracking = bnlx.tracking[bnlx.tracking$FPKM>0,]

high.bnlx = bnlx.tracking[bnlx.tracking$coverage>50,]
high.shr = shr.tracking[shr.tracking$coverage>50,]

high.bnlx$transcript_id = gsub("CUFF","BNLx",high.bnlx$tracking_id)
high.shr$transcript_id = gsub("CUFF","SHR",high.shr$tracking_id)

high.combined = merge(high.bnlx,high.shr,by=c("transcript_id","locus","length"),all=TRUE)
high.combined = high.combined[order(high.combined$locus),c("transcript_id","locus","length","coverage.x","FPKM.x","coverage.y","FPKM.y")]

high.combined = high.combined[high.combined$length>200,]

table(grepl("ENS",high.combined$transcript_id))

shr.transcripts$transcript_id = gsub("CUFF","SHR",shr.transcripts$transcript_id)
bnlx.transcripts$transcript_id = gsub("CUFF","BNLx",bnlx.transcripts$transcript_id)

bnlx.bed = bnlx.transcripts[bnlx.transcripts$transcript_id %in% high.combined$transcript_id[grep("BNLx",high.combined$transcript_id)],]
bnlx.bed$V6 = 1000

shr.bed = shr.transcripts[shr.transcripts$transcript_id %in% high.combined$transcript_id[grep("SHR",high.combined$transcript_id)],]
shr.bed$V6 = 1000

bnlx.bed = bnlx.bed[,c("V1","V4","V5","V6","V7","transcript_id")]
shr.bed = shr.bed[,c("V1","V4","V5","V6","V7","transcript_id")]

write.table(shr.bed,file="RNA-Seq.Brain.polyA/reconstruction/tmp/shr.bed",sep="\t",row.names=FALSE,col.names=FALSE,quote=FALSE)
write.table(bnlx.bed,file="RNA-Seq.Brain.polyA/reconstruction/tmp/bnlx.bed",sep="\t",row.names=FALSE,col.names=FALSE,quote=FALSE)

```

3.  Look for overlap between transcripts within the combined BED file  
-----------------------------------------
BEDtools version = bedtools v2.16.2-zip-5db7ff9


```
export PATH=/usr/local/bedtools2/bin:$PATH
 
cd /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Brain.polyA/reconstruction/tmp
intersectBed -a shr.bed -b bnlx.bed -wo > overlap.polyA.byStrain.txt
```

4.  Identify novel transcripts that were identified in both BN-Lx and SHR 
------------------------------------------------------
Two transcripts were "merged" into one transcripts if:  
* they are both assigned to the same strand or one/both did not have a strand designation
* they were identified as novel in opposite strains (one from SHR and one from BNLx)
* if 1) all exon starts and exon stops matched OR 2) all exon junctions matched, transcription start and stop sites could differ OR 3) two one-exon transcripts with transcription start sites within 100 bp of each other and transcription stop sites within 100 bp of each other

```{r,eval=FALSE}
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BNLx.SHR")

shr = read.table(file = "RNA-Seq.Brain.polyA/reconstruction/SHR.Brain.PolyA.ensemblGuided/transcripts.gtf",sep="\t",header=FALSE)
shr = gtfFormat(shr)
shr$transcript_id = gsub("CUFF","SHR",shr$transcript_id)
shr.gtf <- shr[(shr$transcript_id %in% high.combined$transcript_id[!grepl("ENS",high.combined$transcript_id)]) & shr$V3=="exon",]
shr.gtf <- shr.gtf[order(shr.gtf$transcript_id,shr.gtf$V4,shr.gtf$V5),]


bnlx = read.table(file = "RNA-Seq.Brain.polyA/reconstruction/BNLx.Brain.PolyA.ensemblGuided/transcripts.gtf",sep="\t",header=FALSE)
bnlx = gtfFormat(bnlx)
bnlx$transcript_id = gsub("CUFF","BNLx",bnlx$transcript_id)
bnlx.gtf <- bnlx[(bnlx$transcript_id %in% high.combined$transcript_id[!grepl("ENS",high.combined$transcript_id)]) & bnlx$V3=="exon",]
bnlx.gtf <- bnlx.gtf[order(bnlx.gtf$transcript_id,bnlx.gtf$V4,bnlx.gtf$V5),]

ens.gtf = shr[(shr$transcript_id %in% high.combined$transcript_id[grepl("ENS",high.combined$transcript_id)]) & shr$V3=="exon",]

gtf = rbind(shr.gtf,bnlx.gtf,ens.gtf)
gtf <- gtf[order(gtf$transcript_id,gtf$V4,gtf$V5),]

length(unique(gtf$transcript_id))

##  create one per transcript ##
start <- aggregate(gtf$V4,by=list(gtf$transcript_id),function(a) paste(a,collapse=","))
stop <- aggregate(gtf$V5,by=list(gtf$transcript_id),function(a) paste(a,collapse=","))
transcripts <- gtf[!duplicated(gtf[,c("transcript_id","V1")]),c("transcript_id","V1","V6","V7")]
onePerTrans <- merge(start,stop,by=1)
onePerTrans <- merge(transcripts,onePerTrans,by=1)

onePerTrans$exonNum = unlist(lapply(strsplit(onePerTrans$x.x,split=",",fixed=TRUE),length))
onePerTrans$start <- as.integer(unlist(lapply(strsplit(onePerTrans$x.x,split=",",fixed=TRUE),function(a) min(as.numeric(a)))))
onePerTrans$stop <- as.integer(unlist(lapply(strsplit(onePerTrans$x.y,split=",",fixed=TRUE),function(a) max(as.numeric(a)))))

#####################
overlap <- read.table(file="RNA-Seq.Brain.polyA/reconstruction/tmp/overlap.polyA.byStrain.txt",sep="\t",header=FALSE)
colnames(overlap) <- c(paste(c("chr","start","stop","V5","strand","transcript_id"),rep(c("SHR","BNLx"),each=6),sep="."),"overlap")

## add exon information to overlaps
setSHR = setBNLx = onePerTrans[,c("transcript_id","x.x","x.y","exonNum")]
colnames(setSHR) = c("transcript_id.SHR","exonStarts.SHR","exonStops.SHR","exonNum.SHR")
colnames(setBNLx) = c("transcript_id.BNLx","exonStarts.BNLx","exonStops.BNLx","exonNum.BNLx")

overlap = merge(overlap,setSHR,by="transcript_id.SHR")
overlap = merge(overlap,setBNLx,by="transcript_id.BNLx")

overlap$ID = c(1:nrow(overlap))

##  identify exon junction

findJunct <- function(starts,stops){
  start <- strsplit(starts,split=",",fixed=TRUE)[[1]]
  stop <- strsplit(stops,split=",",fixed=TRUE)[[1]]
  junct <- NA
  if(length(start)>1) junct <- paste(paste(stop[-length(stop)],start[-1],sep="//"),collapse=",")
  return(junct)
  }
  
overlap$exonJunct.SHR = apply(overlap[,c("exonStarts.SHR","exonStops.SHR")],1,function(a) findJunct(a[1],a[2]))
overlap$exonJunct.BNLx = apply(overlap[,c("exonStarts.BNLx","exonStops.BNLx")],1,function(a) findJunct(a[1],a[2]))

## remove overlap that is on the opposite strand
overlap <- overlap[overlap$strand.SHR=="." | overlap$strand.BNLx=="." | overlap$strand.SHR==overlap$strand.BNLx,]

## remove overlap where coding regions don't overlap  ##
intronCheck =c()
for(i in 1:nrow(overlap)){
  input = overlap[i,]
  out = sum(sum(unlist(lapply(strsplit(strsplit(input[1,"exonJunct.BNLx"],split=",")[[1]],split="//"),function(a) input[1,"start.SHR"]>a[1] & input[1,"stop.SHR"]<a[2]))),sum(unlist(lapply(strsplit(strsplit(input[1,"exonJunct.SHR"],split=",")[[1]],split="//"),function(a) input[1,"start.BNLx"]>a[1] & input[1,"stop.BNLx"]<a[2]))),na.rm=TRUE)
  intronCheck = c(intronCheck,out)
  }
  
overlap = overlap[intronCheck==0,]

## transcripts without any overlap - 3,669 transcripts
noOverlap = onePerTrans[!(onePerTrans$transcript_id %in% unique(c(overlap$transcript_id.SHR,overlap$transcript_id.BNLx))),]

##  perfect overlap - 50 transcripts
perfect = overlap[overlap$exonStarts.SHR==overlap$exonStarts.BNLx & overlap$exonStops.SHR==overlap$exonStops.BNLx,]

##  identify exon junction matches - 1,567 transcripts
reduced = overlap[!(overlap$ID %in% perfect$ID),]
exonJunctMatch <- reduced[!is.na(reduced$exonJunct.SHR) & !is.na(reduced$exonJunct.BNLx) & reduced$exonJunct.SHR==reduced$exonJunct.BNLx,]

##  examine one-exon transcripts - 697 transcripts
reduced = overlap[!(overlap$ID %in% c(perfect$ID,exonJunctMatch$ID)),]
oneExons = reduced[reduced$exonNum.SHR==1 & reduced$exonNum.BNLx==1,]

closeMatch.oneExon = oneExons[abs(oneExons$start.SHR - oneExons$start.BNLx)<100 & abs(oneExons$stop.SHR - oneExons$stop.BNLx)<100, ]
sum(duplicated(closeMatch.oneExon$transcript_id.BNLx))

##  Check for overlap between renamed  ##
matched = as.data.frame(rbind(perfect,exonJunctMatch,closeMatch.oneExon))
sum(duplicated(matched$transcript_id.BNLx))
sum(duplicated(matched$transcript_id.SHR))


## Examine Overlap ##
overlap.v2 = overlap[!(overlap$transcript_id.BNLx %in% matched$transcript_id.BNLx) & !(overlap$transcript_id.SHR %in% matched$transcript_id.SHR),]


##  create new transcript for matched transcripts - 1,945 matches
matched$transcript_id = paste("matched",c(1:nrow(matched)),sep=".")
matched$gene_id = NA
matched$V1 = matched$chr.SHR
matched$V6 = 1000
matched$V7 = matched$strand.SHR
matched$start = apply(matched[,c("start.BNLx","start.SHR")],1,min)
matched$stop = apply(matched[,c("stop.BNLx","stop.SHR")],1,max)
matched$exonNum = matched$exonNum.SHR

matched$x.x = NA
matched$x.x[matched$start==matched$start.SHR] = matched$exonStarts.SHR[matched$start==matched$start.SHR]
matched$x.x[matched$start==matched$start.BNLx] = matched$exonStarts.BNLx[matched$start==matched$start.BNLx]

matched$x.y = NA
matched$x.y[matched$stop==matched$stop.SHR] = matched$exonStops.SHR[matched$stop==matched$stop.SHR]
matched$x.y[matched$stop==matched$stop.BNLx] = matched$exonStops.BNLx[matched$stop==matched$stop.BNLx]

matched.toAdd = matched[,colnames(onePerTrans)]

new.transcripts = onePerTrans[!(onePerTrans$transcript_id %in% c(matched$transcript_id.SHR,matched$transcript_id.BNLx)),]
new.transcripts = rbind(new.transcripts,matched.toAdd)

write.table(new.transcripts[,c("V1","start","stop","transcript_id","V6","V7")],file="RNA-Seq.Brain.polyA/reconstruction/tmp/combined.brain.polyA.10Jul14.bed",sep="\t",quote=FALSE,row.names=FALSE,col.names=FALSE)
```
