Initial Quality Control of Liver RNA-Seq Reads From Beckman (received 3/25/14)
=========================
* 2 liver ribosomal RNA depleted total RNA samples, BN-Lx3 and SHR1
* synthetic spikes included Mix 1 in SHR1 and Mix 2 in BN-Lx3
* 2X100 paired end reads using the stranded protocol


1. Unzip FASTQ files - DONE
----------------------------
```
cd /data/Users/hi-seq/BNLx.SHR.Liver.totalRNA.Beckman.25Mar14/rawReads
gzip -d *.gz
```

2. Determine number of reads sent for each sample - DONE
----------------------------------------------------------
```
##  Determine Number of Reads Generated - number of rows divided by 4  ##
wc -l *.fastq > /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Liver/data/numberReads.BNLx.SHR.Liver.25Mar14.txt
```

```{r}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BNLx.SHR/RNA-Seq.Liver/totalRNA.Beckman/data/")
rawCounts = read.table(file="numberReads.BNLx.SHR.Liver.25Mar14.txt",sep=" ",header=FALSE,fill=TRUE)
rawCounts = rawCounts[is.na(rawCounts$V2),]
rawCounts$readFrag = as.numeric(rawCounts$V3)/4
```

BNLx3 = `r prettyNum(sum(rawCounts$readFrag[grep("BN",rawCounts$V4)]),big.mark=",",scientific=FALSE)` read fragments (`r prettyNum(sum(rawCounts$readFrag[grep("BN",rawCounts$V4)])/2,big.mark=",",scientific=FALSE)` paired reads)  
SHR1 = `r prettyNum(sum(rawCounts$readFrag[grep("SHR",rawCounts$V4)]),big.mark=",",scientific=FALSE)` read fragments (`r prettyNum(sum(rawCounts$readFrag[grep("SHR",rawCounts$V4)])/2,big.mark=",",scientific=FALSE)` paired reads)


3. Trim Reads for Adaptors and for Quality - DONE
--------------------------------------------
```
export PATH=$HOME/bin:$PATH
export PYTHONPATH=$HOME/bin:$PYTHONPATH

cd /data/Users/hi-seq/BNLx.SHR.Liver.totalRNA.Beckman.25Mar14/

trim_galore --paired --stringency 3 -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -q 20 -o ./trimmedReads/ ./rawReads/BN_LX-3_CTTGTA_R1.fastq ./rawReads/BN_LX-3_CTTGTA_R2.fastq
trim_galore --paired --stringency 3 -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -q 20 -o ./trimmedReads/ ./rawReads/SHR_H-1_CGATGT_R1.fastq ./rawReads/SHR_H-1_CGATGT_R2.fastq
```

4. Characterizing Trimmed Reads - DONE
----------------------------------
```
export PATH=$HOME/bin:$PATH
export PYTHONPATH=$HOME/bin:$PYTHONPATH

cd /data/Users/hi-seq/BNLx.SHR.Liver.totalRNA.Beckman.25Mar14/trimmedReads
awk  '/@HWI/ {getline; print length($0)}' BN*.fq | awk '{sum+=$1} END { print "BNLx3","\t",sum/NR,"\t",NR}' > /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Liver/data/trimmedInfo.BNLx.SHR.Liver.25Mar14.txt
awk  '/@HWI/ {getline; print length($0)}' SHR*.fq | awk '{sum+=$1} END { print "SHR1","\t",sum/NR,"\t",NR}' >> /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Liver/data/trimmedInfo.BNLx.SHR.Liver.25Mar14.txt
```

```{r}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/BNLx.SHR/RNA-Seq.Liver/totalRNA.Beckman/data/")
trimmed = read.table(file="trimmedInfo.BNLx.SHR.Liver.25Mar14.txt",sep="\t",header=FALSE,row.names=1)
```

BNLx3:
------
Average read fragment length = `r round(trimmed["BNLx3","V2"],2)`  
Number of read fragments that remain after trimming = `r prettyNum(trimmed["BNLx3","V3"],big.mark=",")`

SHR1:
------
Average read fragment length = `r round(trimmed["SHR1","V2"],2)`  
Number of read fragments that remain after trimming = `r prettyNum(trimmed["SHR1","V3"],big.mark=",")`


5.  Generate Strain-Specific Genomes Including Spike-In Sequences - DONE
-------------------------------------------
```
cd /data/Tabastore3/LauraS/index
cat /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq/index/BNLX_rn5_Genome.fa /data/Tabastore3/LauraS/BxH.HxB.Rats/RNA-Seq/spikeSource/ERCC92.fa > BNLx_rn5_wSpikes.fa
bowtie2-build BNLx_rn5_wSpikes.fa BNLx_rn5_wSpikes

cd /data/Tabastore3/LauraS/BxH.HxB.Rats/RNA-Seq/index/
cp SHR_rn5_Genome.wSpikes* /data/Tabastore3/LauraS/index
```

6. Alignment of Raw Reads to Strain-Specific Genomes
------------------------------------

```
qsub -q smp-q /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/programs/alignment.14Apr14.sh
```

### Code from alignment.14Apr14.sh
```
#!/bin/bash
module add bio/bowtie2
module add bio/tophat
tophat2 --library-type fr-firststrand -o /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/alignedReads/BNLx3 -p 10 /home/saba/index/BNLx_rn5_wSpikes /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/trimmedReads/BN_LX-3_CTTGTA_R1_val_1.fq /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/trimmedReads/BN_LX-3_CTTGTA_R2_val_2.fq
tophat2 --library-type fr-firststrand -o /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/alignedReads/SHR1 -p 10 /home/saba/index/SHR_rn5_wSpikes /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/trimmedReads/SHR_H-1_CGATGT_R1_val_1.fq /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/trimmedReads/SHR_H-1_CGATGT_R2_val_2.fq
```


7.  Characterization of Aligned Reads
---------------------------

```{r}
sampleList=c("BNLx3","SHR1")

alignStats = c()
for(i in sampleList){
  x = read.table(file=paste("/Volumes/LauraS/BNLx.SHR/RNA-Seq.Liver/totalRNA.Beckman/alignedReads/",i,"/align_summary.txt",sep=""),sep="\t",header=FALSE)
  sample = i
  numReads = as.numeric(gsub(":","",gsub("Input","",x[2,1])))  
  mappedLeft = as.numeric(gsub(":","",gsub("Mapped","",unlist(lapply(strsplit(x[3,1],split="(",fixed=TRUE),function(a) a[1])))))
  mappedRight = as.numeric(gsub(":","",gsub("Mapped","",unlist(lapply(strsplit(x[7,1],split="(",fixed=TRUE),function(a) a[1])))))
  multiLeft = as.numeric(gsub("of these: ","",unlist(lapply(strsplit(x[4,1],split="(",fixed=TRUE),function(a) a[1]))))
  multiRight = as.numeric(gsub("of these: ","",unlist(lapply(strsplit(x[8,1],split="(",fixed=TRUE),function(a) a[1]))))
  lotsLeft = as.numeric(gsub(" have >20)","",unlist(lapply(strsplit(x[4,1],split="(",fixed=TRUE),function(a) a[3]))))
  lotsRight = as.numeric(gsub(" have >20)","",unlist(lapply(strsplit(x[8,1],split="(",fixed=TRUE),function(a) a[3]))))
  mappedPairs = as.numeric(gsub("Aligned pairs: ","",x[10,1]))  
  multiPairs = as.numeric(gsub("of these: ","",unlist(lapply(strsplit(x[11,1],split="(",fixed=TRUE),function(a) a[1]))))
  discordPairs = as.numeric(gsub("and: ","",unlist(lapply(strsplit(x[12,1],split="(",fixed=TRUE),function(a) a[1]))))

  alignStats = rbind(alignStats,data.frame(sample,numReads,mappedLeft,mappedRight,multiLeft,multiRight,lotsLeft,lotsRight,mappedPairs,multiPairs,discordPairs))
}

alignTable = data.frame(t(alignStats))
colnames(alignTable) = alignTable[1,]
alignTable = alignTable[-1,]
alignTable$BNLx3.pct = as.numeric(alignTable$BNLx3)/as.numeric(alignTable["numReads","BNLx3"])
alignTable$SHR1.pct = as.numeric(alignTable$SHR1)/as.numeric(alignTable["numReads","SHR1"])

## format numbers
alignTable$BNLx3 = prettyNum(alignTable$BNLx3,big.mark=",",scientific=FALSE)
alignTable$SHR1 = prettyNum(alignTable$SHR1,big.mark=",",scientific=FALSE)
alignTable$BNLx3.pct = paste(sprintf("%.2f", round(alignTable$BNLx3.pct*100,2)),"%",sep="")
alignTable$SHR1.pct = paste(sprintf("%.2f", round(alignTable$SHR1.pct*100,2)),"%",sep="")

## add labels
labels = read.table(file="/Volumes/LauraS/NextGenSeq/misc/alignLabels.txt",sep="\t",header=TRUE)
alignTable = merge(labels,alignTable,by.x="var",by.y=0)
alignTable = alignTable[order(alignTable$orderNum),]
alignTable = alignTable[,c("Label","BNLx3","BNLx3.pct","SHR1","SHR1.pct")]

````

### Statistics on Alignment to Genome
```{r, results='asis',echo=FALSE}
kable(alignTable,align=rep("c",ncol(alignTable)),row.names=FALSE)
```

8.  Examine Spike-In Expression 
-----------------------

```
qsub -q smp-q /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/programs/quantifySpikes.sh
```

9.  Create BigWig Files
-------------------
```
qsub -q smp /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/programs/testBigWig.sh
```

```
track type=bigWig name="Beckman.plus" description="BNLx3 (Beckman) - PLUS" bigDataUrl=http://ucsc:JU7etr5t@phenogen.ucdenver.edu/ucsc/BNLx3.plus.v2.bw
track type=bigWig name="Beckman.minus" description="BNLx3 (Beckman) - MINUS" bigDataUrl=http://ucsc:JU7etr5t@phenogen.ucdenver.edu/ucsc/BNLx3.minus.bw


