<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>Initial Quality Control of Liver RNA-Seq Reads From Beckman (received 3/25/14)</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: rgb(88, 72, 246)
   }

   pre .number {
     color: rgb(0, 0, 205);
   }

   pre .comment {
     color: rgb(76, 136, 107);
   }

   pre .keyword {
     color: rgb(0, 0, 255);
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: rgb(3, 106, 7);
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>




</head>

<body>
<h1>Initial Quality Control of Liver RNA-Seq Reads From Beckman (received 3/25/14)</h1>

<ul>
<li>2 liver ribosomal RNA depleted total RNA samples, BN-Lx3 and SHR1</li>
<li>synthetic spikes included Mix 1 in SHR1 and Mix 2 in BN-Lx3</li>
<li>2X100 paired end reads using the stranded protocol</li>
</ul>

<h2>1. Unzip FASTQ files - DONE</h2>

<pre><code>cd /data/Users/hi-seq/BNLx.SHR.Liver.totalRNA.Beckman.25Mar14/rawReads
gzip -d *.gz
</code></pre>

<h2>2. Determine number of reads sent for each sample - DONE</h2>

<pre><code>##  Determine Number of Reads Generated - number of rows divided by 4  ##
wc -l *.fastq &gt; /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Liver/data/numberReads.BNLx.SHR.Liver.25Mar14.txt
</code></pre>

<pre><code class="r">rm(list = ls())
options(stringsAsFactors = FALSE)
setwd(&quot;/Volumes/LauraS/BNLx.SHR/RNA-Seq.Liver/totalRNA.Beckman/data/&quot;)
rawCounts = read.table(file = &quot;numberReads.BNLx.SHR.Liver.25Mar14.txt&quot;, sep = &quot; &quot;, 
    header = FALSE, fill = TRUE)
rawCounts = rawCounts[is.na(rawCounts$V2), ]
rawCounts$readFrag = as.numeric(rawCounts$V3)/4
</code></pre>

<p>BNLx3 = 150,598,054 read fragments (75,299,027 paired reads)<br/>
SHR1 = 153,528,402 read fragments (76,764,201 paired reads)</p>

<h2>3. Trim Reads for Adaptors and for Quality - DONE</h2>

<pre><code>export PATH=$HOME/bin:$PATH
export PYTHONPATH=$HOME/bin:$PYTHONPATH

cd /data/Users/hi-seq/BNLx.SHR.Liver.totalRNA.Beckman.25Mar14/

trim_galore --paired --stringency 3 -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -q 20 -o ./trimmedReads/ ./rawReads/BN_LX-3_CTTGTA_R1.fastq ./rawReads/BN_LX-3_CTTGTA_R2.fastq
trim_galore --paired --stringency 3 -a GATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -q 20 -o ./trimmedReads/ ./rawReads/SHR_H-1_CGATGT_R1.fastq ./rawReads/SHR_H-1_CGATGT_R2.fastq
</code></pre>

<h2>4. Characterizing Trimmed Reads - DONE</h2>

<pre><code>export PATH=$HOME/bin:$PATH
export PYTHONPATH=$HOME/bin:$PYTHONPATH

cd /data/Users/hi-seq/BNLx.SHR.Liver.totalRNA.Beckman.25Mar14/trimmedReads
awk  &#39;/@HWI/ {getline; print length($0)}&#39; BN*.fq | awk &#39;{sum+=$1} END { print &quot;BNLx3&quot;,&quot;\t&quot;,sum/NR,&quot;\t&quot;,NR}&#39; &gt; /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Liver/data/trimmedInfo.BNLx.SHR.Liver.25Mar14.txt
awk  &#39;/@HWI/ {getline; print length($0)}&#39; SHR*.fq | awk &#39;{sum+=$1} END { print &quot;SHR1&quot;,&quot;\t&quot;,sum/NR,&quot;\t&quot;,NR}&#39; &gt;&gt; /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq.Liver/data/trimmedInfo.BNLx.SHR.Liver.25Mar14.txt
</code></pre>

<pre><code class="r">rm(list = ls())
options(stringsAsFactors = FALSE)
setwd(&quot;/Volumes/LauraS/BNLx.SHR/RNA-Seq.Liver/totalRNA.Beckman/data/&quot;)
trimmed = read.table(file = &quot;trimmedInfo.BNLx.SHR.Liver.25Mar14.txt&quot;, sep = &quot;\t&quot;, 
    header = FALSE, row.names = 1)
</code></pre>

<h2>BNLx3:</h2>

<p>Average read fragment length = 97.87<br/>
Number of read fragments that remain after trimming = 149,653,216</p>

<h2>SHR1:</h2>

<p>Average read fragment length = 96.8<br/>
Number of read fragments that remain after trimming = 152,303,118</p>

<h2>5.  Generate Strain-Specific Genomes Including Spike-In Sequences - DONE</h2>

<pre><code>cd /data/Tabastore3/LauraS/index
cat /data/Tabastore3/LauraS/BNLx.SHR/RNA-Seq/index/BNLX_rn5_Genome.fa /data/Tabastore3/LauraS/BxH.HxB.Rats/RNA-Seq/spikeSource/ERCC92.fa &gt; BNLx_rn5_wSpikes.fa
bowtie2-build BNLx_rn5_wSpikes.fa BNLx_rn5_wSpikes

cd /data/Tabastore3/LauraS/BxH.HxB.Rats/RNA-Seq/index/
cp SHR_rn5_Genome.wSpikes* /data/Tabastore3/LauraS/index
</code></pre>

<h2>6. Alignment of Raw Reads to Strain-Specific Genomes</h2>

<pre><code>qsub -q smp-q /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/programs/alignment.14Apr14.sh
</code></pre>

<h3>Code from alignment.14Apr14.sh</h3>

<pre><code>#!/bin/bash
module add bio/bowtie2
module add bio/tophat
tophat2 --library-type fr-firststrand -o /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/alignedReads/BNLx3 -p 10 /home/saba/index/BNLx_rn5_wSpikes /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/trimmedReads/BN_LX-3_CTTGTA_R1_val_1.fq /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/trimmedReads/BN_LX-3_CTTGTA_R2_val_2.fq
tophat2 --library-type fr-firststrand -o /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/alignedReads/SHR1 -p 10 /home/saba/index/SHR_rn5_wSpikes /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/trimmedReads/SHR_H-1_CGATGT_R1_val_1.fq /home/saba/BNLx.SHR.Liver.totalRNA.Beckman/trimmedReads/SHR_H-1_CGATGT_R2_val_2.fq
</code></pre>

<h2>7.  Characterization of Aligned Reads</h2>

<pre><code class="r">sampleList = c(&quot;BNLx3&quot;, &quot;SHR1&quot;)

alignStats = c()
for (i in sampleList) {
    x = read.table(file = paste(&quot;/Volumes/LauraS/BNLx.SHR/RNA-Seq.Liver/totalRNA.Beckman/alignedReads/&quot;, 
        i, &quot;/align_summary.txt&quot;, sep = &quot;&quot;), sep = &quot;\t&quot;, header = FALSE)
    sample = i
    numReads = as.numeric(gsub(&quot;:&quot;, &quot;&quot;, gsub(&quot;Input&quot;, &quot;&quot;, x[2, 1])))
    mappedLeft = as.numeric(gsub(&quot;:&quot;, &quot;&quot;, gsub(&quot;Mapped&quot;, &quot;&quot;, unlist(lapply(strsplit(x[3, 
        1], split = &quot;(&quot;, fixed = TRUE), function(a) a[1])))))
    mappedRight = as.numeric(gsub(&quot;:&quot;, &quot;&quot;, gsub(&quot;Mapped&quot;, &quot;&quot;, unlist(lapply(strsplit(x[7, 
        1], split = &quot;(&quot;, fixed = TRUE), function(a) a[1])))))
    multiLeft = as.numeric(gsub(&quot;of these: &quot;, &quot;&quot;, unlist(lapply(strsplit(x[4, 
        1], split = &quot;(&quot;, fixed = TRUE), function(a) a[1]))))
    multiRight = as.numeric(gsub(&quot;of these: &quot;, &quot;&quot;, unlist(lapply(strsplit(x[8, 
        1], split = &quot;(&quot;, fixed = TRUE), function(a) a[1]))))
    lotsLeft = as.numeric(gsub(&quot; have &gt;20)&quot;, &quot;&quot;, unlist(lapply(strsplit(x[4, 
        1], split = &quot;(&quot;, fixed = TRUE), function(a) a[3]))))
    lotsRight = as.numeric(gsub(&quot; have &gt;20)&quot;, &quot;&quot;, unlist(lapply(strsplit(x[8, 
        1], split = &quot;(&quot;, fixed = TRUE), function(a) a[3]))))
    mappedPairs = as.numeric(gsub(&quot;Aligned pairs: &quot;, &quot;&quot;, x[10, 1]))
    multiPairs = as.numeric(gsub(&quot;of these: &quot;, &quot;&quot;, unlist(lapply(strsplit(x[11, 
        1], split = &quot;(&quot;, fixed = TRUE), function(a) a[1]))))
    discordPairs = as.numeric(gsub(&quot;and: &quot;, &quot;&quot;, unlist(lapply(strsplit(x[12, 
        1], split = &quot;(&quot;, fixed = TRUE), function(a) a[1]))))

    alignStats = rbind(alignStats, data.frame(sample, numReads, mappedLeft, 
        mappedRight, multiLeft, multiRight, lotsLeft, lotsRight, mappedPairs, 
        multiPairs, discordPairs))
}

alignTable = data.frame(t(alignStats))
colnames(alignTable) = alignTable[1, ]
alignTable = alignTable[-1, ]
alignTable$BNLx3.pct = as.numeric(alignTable$BNLx3)/as.numeric(alignTable[&quot;numReads&quot;, 
    &quot;BNLx3&quot;])
alignTable$SHR1.pct = as.numeric(alignTable$SHR1)/as.numeric(alignTable[&quot;numReads&quot;, 
    &quot;SHR1&quot;])

## format numbers
alignTable$BNLx3 = prettyNum(alignTable$BNLx3, big.mark = &quot;,&quot;, scientific = FALSE)
alignTable$SHR1 = prettyNum(alignTable$SHR1, big.mark = &quot;,&quot;, scientific = FALSE)
alignTable$BNLx3.pct = paste(sprintf(&quot;%.2f&quot;, round(alignTable$BNLx3.pct * 100, 
    2)), &quot;%&quot;, sep = &quot;&quot;)
alignTable$SHR1.pct = paste(sprintf(&quot;%.2f&quot;, round(alignTable$SHR1.pct * 100, 
    2)), &quot;%&quot;, sep = &quot;&quot;)

## add labels
labels = read.table(file = &quot;/Volumes/LauraS/NextGenSeq/misc/alignLabels.txt&quot;, 
    sep = &quot;\t&quot;, header = TRUE)
alignTable = merge(labels, alignTable, by.x = &quot;var&quot;, by.y = 0)
alignTable = alignTable[order(alignTable$orderNum), ]
alignTable = alignTable[, c(&quot;Label&quot;, &quot;BNLx3&quot;, &quot;BNLx3.pct&quot;, &quot;SHR1&quot;, &quot;SHR1.pct&quot;)]
</code></pre>

<h3>Statistics on Alignment to Genome</h3>

<table><thead>
<tr>
<th align="center">Label</th>
<th align="center">BNLx3</th>
<th align="center">BNLx3.pct</th>
<th align="center">SHR1</th>
<th align="center">SHR1.pct</th>
</tr>
</thead><tbody>
<tr>
<td align="center">number of paired-end reads</td>
<td align="center">74,826,608</td>
<td align="center">100.00%</td>
<td align="center">76,151,559</td>
<td align="center">100.00%</td>
</tr>
<tr>
<td align="center">number of left reads mapped to genome</td>
<td align="center">63,339,834</td>
<td align="center">84.65%</td>
<td align="center">61,153,559</td>
<td align="center">80.31%</td>
</tr>
<tr>
<td align="center">number of right reads mapped to genome</td>
<td align="center">63,697,146</td>
<td align="center">85.13%</td>
<td align="center">61,804,327</td>
<td align="center">81.16%</td>
</tr>
<tr>
<td align="center">number of mapped left reads with multiple alignments</td>
<td align="center">10,723,581</td>
<td align="center">14.33%</td>
<td align="center">11,869,711</td>
<td align="center">15.59%</td>
</tr>
<tr>
<td align="center">number of mapped right reads with multiple alignments</td>
<td align="center">10,849,654</td>
<td align="center">14.50%</td>
<td align="center">12,118,110</td>
<td align="center">15.91%</td>
</tr>
<tr>
<td align="center">number of mapped left reads with more than 20 alignments</td>
<td align="center">517,608</td>
<td align="center">0.69%</td>
<td align="center">289,346</td>
<td align="center">0.38%</td>
</tr>
<tr>
<td align="center">number of mapped right reads with more than 20 alignments</td>
<td align="center">520,032</td>
<td align="center">0.69%</td>
<td align="center">289,713</td>
<td align="center">0.38%</td>
</tr>
<tr>
<td align="center">number of mapped read pairs</td>
<td align="center">62,008,548</td>
<td align="center">82.87%</td>
<td align="center">59,377,570</td>
<td align="center">77.97%</td>
</tr>
<tr>
<td align="center">number of mapped read pairs with multiple alignments</td>
<td align="center">10,281,516</td>
<td align="center">13.74%</td>
<td align="center">11,342,303</td>
<td align="center">14.89%</td>
</tr>
<tr>
<td align="center">number of mapped read pairs with discordant alignments</td>
<td align="center">1,035,103</td>
<td align="center">1.38%</td>
<td align="center">850,214</td>
<td align="center">1.12%</td>
</tr>
</tbody></table>

<h2>8.  Examine Spike-In Expression </h2>

<pre><code>##  NOT YET UPDATED

cd /data/Tabastore3/LauraS/BxH.HxB.Rats/RNA-Seq/
samtools view ./alignedReads/HXB25_1/accepted_hits.bam | awk &#39;$3!~&quot;chr&quot;&#39; - &gt; ./alignedReads/HXB25_1/spikeOnly.sam 
samtools view ./alignedReads/HXB25_2/accepted_hits.bam | awk &#39;$3!~&quot;chr&quot;&#39; - &gt; ./alignedReads/HXB25_2/spikeOnly.sam 
samtools view ./alignedReads/HXB25_3/accepted_hits.bam | awk &#39;$3!~&quot;chr&quot;&#39; - &gt; ./alignedReads/HXB25_3/spikeOnly.sam 

samtools view ./alignedReads/HXB2_1/accepted_hits.bam | awk &#39;$3!~&quot;chr&quot;&#39; - &gt; ./alignedReads/HXB2_1/spikeOnly.sam 
samtools view ./alignedReads/HXB2_2/accepted_hits.bam | awk &#39;$3!~&quot;chr&quot;&#39; - &gt; ./alignedReads/HXB2_2/spikeOnly.sam 
samtools view ./alignedReads/HXB2_3/accepted_hits.bam | awk &#39;$3!~&quot;chr&quot;&#39; - &gt; ./alignedReads/HXB2_3/spikeOnly.sam 


cd /data/Tabastore3/LauraS/BxH.HxB.Rats/RNA-Seq/
cufflinks -p 2 -o ./spikeCounts/HXB25_1 -b ./index/SHR_rn5_Genome.wSpikes.fa -u -G ./spikeSource/ERCC92.none.gtf ./alignedReads/HXB25_1/spikeOnly.sam
cufflinks -p 2 -o ./spikeCounts/HXB25_2 -b ./index/SHR_rn5_Genome.wSpikes.fa -u -G ./spikeSource/ERCC92.none.gtf ./alignedReads/HXB25_2/spikeOnly.sam
cufflinks -p 2 -o ./spikeCounts/HXB25_3 -b ./index/SHR_rn5_Genome.wSpikes.fa -u -G ./spikeSource/ERCC92.none.gtf ./alignedReads/HXB25_3/spikeOnly.sam

cufflinks -p 2 -o ./spikeCounts/HXB2_1 -b ./index/SHR_rn5_Genome.wSpikes.fa -u -G ./spikeSource/ERCC92.none.gtf ./alignedReads/HXB2_1/spikeOnly.sam
cufflinks -p 2 -o ./spikeCounts/HXB2_2 -b ./index/SHR_rn5_Genome.wSpikes.fa -u -G ./spikeSource/ERCC92.none.gtf ./alignedReads/HXB2_2/spikeOnly.sam
cufflinks -p 2 -o ./spikeCounts/HXB2_3 -b ./index/SHR_rn5_Genome.wSpikes.fa -u -G ./spikeSource/ERCC92.none.gtf ./alignedReads/HXB2_3/spikeOnly.sam


samtools view -bT ./index/rn5.wSpikes.fa ./alignedReads/HXB25_1/spikeOnly.sam | genomeCoverageBed -d -split -strand - -ibam stdin | sort - &gt; ./spikeCounts/spikeInCoverage.HXB25_1.txt
samtools view -bT ./index/rn5.wSpikes.fa ./alignedReads/HXB25_2/spikeOnly.sam | genomeCoverageBed -d -split -strand - -ibam stdin | sort - &gt; ./spikeCounts/spikeInCoverage.HXB25_2.txt
samtools view -bT ./index/rn5.wSpikes.fa ./alignedReads/HXB25_3/spikeOnly.sam | genomeCoverageBed -d -split -strand - -ibam stdin | sort - &gt; ./spikeCounts/spikeInCoverage.HXB25_3.txt


awk &#39;{if($2==16) print $3}&#39; ./alignedReads/HXB25_1/spikeOnly.sam | sort - | uniq -c &gt; ./spikeCounts/readCounts.spikes.HXB25_1.txt
awk &#39;{if($2==16) print $3}&#39; ./alignedReads/HXB25_2/spikeOnly.sam | sort - | uniq -c &gt; ./spikeCounts/readCounts.spikes.HXB25_2.txt
awk &#39;{if($2==16) print $3}&#39; ./alignedReads/HXB25_3/spikeOnly.sam | sort - | uniq -c &gt; ./spikeCounts/readCounts.spikes.HXB25_3.txt

awk &#39;{if($2==16) print $3}&#39; ./alignedReads/HXB2_1/spikeOnly.sam | sort - | uniq -c &gt; ./spikeCounts/readCounts.spikes.HXB2_1.txt
awk &#39;{if($2==16) print $3}&#39; ./alignedReads/HXB2_2/spikeOnly.sam | sort - | uniq -c &gt; ./spikeCounts/readCounts.spikes.HXB2_2.txt
awk &#39;{if($2==16) print $3}&#39; ./alignedReads/HXB2_3/spikeOnly.sam | sort - | uniq -c &gt; ./spikeCounts/readCounts.spikes.HXB2_3.txt



##  number of aligned reads  ##
samtools view -F 0X100 ./alignedReads/HXB25_1/accepted_hits.bam | wc -l 
samtools view -F 0X100 ./alignedReads/HXB25_2/accepted_hits.bam | wc -l 
samtools view -F 0X100 ./alignedReads/HXB25_3/accepted_hits.bam | wc -l 

</code></pre>

</body>

</html>

