---
title: "RI Panel - Batch 1 (Liver) Total RNA RNA-Seq"
author: "Laura Saba"
date: "July 15, 2015"
output:
  md_document:
    variant: markdown_github
---

RI Panel - Batch 1 (Liver) Total RNA RNA-Seq
========================================================

* 2 samples per strain (SHR, BXH12, HXB13, HXB17, HXB2, HXB25, HXB27, HXB7)
* tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic
* ribosomal RNA depleted total RNA
* 2X100 reads using the stranded protocol
* sequencing done by Beckman Coulter Genomics - received June 5, 2015
* synthetic spike-in were added to each sample 
  * SHR_1, BXH12_1, HXB13_1, HXB17_1, HXB2_1, HXB25_2, HXB27_1, and HXB7_1 have mix 1; SHR_2, BXH12_2, HXB13_2, HXB17_2, HXB2_2, HXB25_1, HXB27_2, and HXB7_2 have mix 2
  * 4 mL of 1:100 diluted spikes were added after RNA extraction but before rRNA-depletion
  * Information regarding the spike-ins can be found here: http://products.invitrogen.com/ivgn/product/4456739

1. Unzip Files - DONE
---------------
```
qsub -q smp -N liver.batch1 /storage/saba/Rat.RI.Liver.totalRNA.batch1/programs/unzip.yucca.sh
```

2. Determine number of reads sent for each sample - DONE
-------------------------------------------------
```
qsub -q smp -N cntRaw /storage/saba/Rat.RI.Liver.totalRNA.batch1/programs/countRawReads.liver.batch1.sh
```

```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
library(knitr)

base.folder = "~/Documents/NextGenSeq/Rat.RI.Liver.totalRNA.batch1/"

rawCounts = read.table(file=paste(base.folder,"data/rawReadCounts.01Jul15.txt",sep=""),sep=" ",header=FALSE,fill=TRUE)

rawCounts$readFrag = as.numeric(rawCounts$V3)
rawCounts$file = unlist(lapply(strsplit(rawCounts$V1,split="/",fixed=TRUE),function(a) a[length(a)]))
rawCounts$strain = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) a[1]))
rawCounts$sampleNum = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) gsub("liver","",a[2])))
rawCounts$sample = paste(rawCounts$strain,rawCounts$sampleNum,sep="_")

readFragments = aggregate(rawCounts$readFrag,by=list(sample=rawCounts$sample),sum)
readFragments$numPairedReads = prettyNum(readFragments$x,big.mark=",",scientific=FALSE)
readFragments$numReadFragments = prettyNum(readFragments$x*2,big.mark=",",scientific=FALSE)

readFragments=readFragments[,colnames(readFragments)!="x"]
forPrint = readFragments[,c("sample","numPairedReads","numReadFragments")]
colnames(forPrint) = c("sample","Number of Paired-End Reads","Number of Read Fragments")
```

Raw Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint,align=rep("c",ncol(readFragments)))
```

Total Number of Paired End Reads: `r prettyNum(sum(rawCounts$readFrag)/2,big.mark=",",scientific=FALSE)`  
Total Number of Read Fragments:  `r prettyNum(sum(rawCounts$readFrag),big.mark=",",scientific=FALSE)`  
Average Number of Paired End Reads Per Sample: `r prettyNum(sum(rawCounts$readFrag/2)/length(unique(rawCounts$sample)),big.mark=",",scientific=FALSE)`  

3. Trim reads for adaptors and for quality - 
---------------

```
qsub -q smp -N trimBatch1 /storage/saba/Rat.RI.Liver.totalRNA.batch1/programs/trimReads.liver.batch1.sh
```

Compress raw reads after trimmed reads become available
```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch1/programs/compressRawReads.sh
```

4. Characterize Trimmed Reads - RUNNING
---------------
```
qsub -q smp -N cntTrimmed1 /storage/saba/Rat.RI.Liver.totalRNA.batch1/programs/countTrimmedReads.liver.batch1.sh
```

5. Align to ribosomal RNA - RUNNING
---------------------------
```
qsub -q smp -N rRNA1 /storage/saba/Rat.RI.Liver.totalRNA.batch1/programs/alignTo.rRNA.liver.totalRNA.batch1.sh
```

