<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>BNLx and SHR Brain ribosomal RNA depleted total RNA</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>BNLx and SHR Brain ribosomal RNA depleted total RNA</h1>

<ul>
<li>6 brain ribosomal RNA depleted total RNA samples, SHR1, SHR2, SHR3, BNLx1, BNLx2, and BNLx3

<ul>
<li>2X100 paired end reads using the unstranded protocol</li>
</ul></li>
</ul>

<h2>1. Unzip raw reads - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/unzip.yucca.sh
</code></pre>

<h2>2. Consolidate reads by sample and determine number of reads sent for each sample </h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/countRawReads.sh
</code></pre>

<h2>Raw Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">90,846,224</td>
<td align="center">181,692,448</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">101,364,795</td>
<td align="center">202,729,590</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">103,455,074</td>
<td align="center">206,910,148</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">98,205,223</td>
<td align="center">196,410,446</td>
</tr>
<tr>
<td align="center">SHR2</td>
<td align="center">99,380,646</td>
<td align="center">198,761,292</td>
</tr>
<tr>
<td align="center">SHR3</td>
<td align="center">151,429,599</td>
<td align="center">302,859,198</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads: 644,681,561<br/>
Total Number of Read Fragments:  1,289,363,122<br/>
Average Number of Paired End Reads Per Sample: 107,446,927  </p>

<h2>3. Trim reads for adaptors and for quality - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/trimReads.yucca.sh
</code></pre>

<h2>4.  Characterize trimmed reads - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/countTrimmedReads.sh
</code></pre>

<h2>Trimmed Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
<th align="center">Average Read Fragment Length After Trimming (first read fragment)</th>
<th align="center">Average Read Fragment Length After Trimming (second read fragment)</th>
<th align="center">Number of Read Fragments After Trimming</th>
<th align="center">Percent of Read Fragments That Remained After Trimming</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">90,846,224</td>
<td align="center">181,692,448</td>
<td align="center">96.7</td>
<td align="center">94.9</td>
<td align="center">175,440,236</td>
<td align="center">96.6%</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">101,364,795</td>
<td align="center">202,729,590</td>
<td align="center">97.7</td>
<td align="center">96.0</td>
<td align="center">189,391,052</td>
<td align="center">93.4%</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">103,455,074</td>
<td align="center">206,910,148</td>
<td align="center">98.1</td>
<td align="center">96.6</td>
<td align="center">203,559,392</td>
<td align="center">98.4%</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">98,205,223</td>
<td align="center">196,410,446</td>
<td align="center">97.7</td>
<td align="center">95.8</td>
<td align="center">190,582,634</td>
<td align="center">97.0%</td>
</tr>
<tr>
<td align="center">SHR2</td>
<td align="center">99,380,646</td>
<td align="center">198,761,292</td>
<td align="center">97.8</td>
<td align="center">96.4</td>
<td align="center">188,637,976</td>
<td align="center">94.9%</td>
</tr>
<tr>
<td align="center">SHR3</td>
<td align="center">151,429,599</td>
<td align="center">302,859,198</td>
<td align="center">97.7</td>
<td align="center">95.9</td>
<td align="center">299,201,504</td>
<td align="center">98.8%</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads After Trimming: 623,406,397<br/>
Total Number of Read Fragments After Trimming:  1,246,812,794<br/>
Average Number of Paired End Reads Per Sample After Trimming: 103,901,066  </p>

<h2>5.  Align trimmed reads to ribosomal RNA - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/alignTo.rRNA.brain.01Jul14.sh
</code></pre>

<p><em>tranfer files to phenogen server</em><br/>
get BNLx1.rRNA/align_summary.txt rRNA.summary.BNLx1.txt<br/>
get BNLx2.rRNA/align_summary.txt rRNA.summary.BNLx2.txt<br/>
get BNLx3.rRNA/align_summary.txt rRNA.summary.BNLx3.txt<br/>
get SHR1.rRNA/align_summary.txt rRNA.summary.SHR1.txt<br/>
get SHR2.rRNA/align_summary.txt rRNA.summary.SHR2.txt<br/>
get SHR3.rRNA/align_summary.txt rRNA.summary.SHR3.txt  </p>

<table><thead>
<tr>
<th align="center">Sample</th>
<th align="center">Number of Paired-End Reads After Trimming</th>
<th align="center">Number of Paired-End Reads with at Least One Fragment Aligned to rRNA</th>
<th align="center">Percent of Paired-End Reads Aligned to rRNA</th>
<th align="center">Number of Paired-End Reads NOT Aligned to rRNA</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BNLx1</td>
<td align="center">87,720,118</td>
<td align="center">1,961,603</td>
<td align="center">2.2%</td>
<td align="center">85,758,515</td>
</tr>
<tr>
<td align="center">BNLx2</td>
<td align="center">94,695,526</td>
<td align="center">1,667,365</td>
<td align="center">1.8%</td>
<td align="center">93,028,161</td>
</tr>
<tr>
<td align="center">BNLx3</td>
<td align="center">101,779,696</td>
<td align="center">1,672,624</td>
<td align="center">1.6%</td>
<td align="center">100,107,072</td>
</tr>
<tr>
<td align="center">SHR1</td>
<td align="center">95,291,317</td>
<td align="center">1,327,152</td>
<td align="center">1.4%</td>
<td align="center">93,964,165</td>
</tr>
<tr>
<td align="center">SHR2</td>
<td align="center">94,318,988</td>
<td align="center">1,371,750</td>
<td align="center">1.5%</td>
<td align="center">92,947,238</td>
</tr>
<tr>
<td align="center">SHR3</td>
<td align="center">149,600,752</td>
<td align="center">3,227,627</td>
<td align="center">2.2%</td>
<td align="center">146,373,125</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads After Eliminating rRNA Aligned Reads: 612,178,276<br/>
Average Number of Paired End Reads Per Sample After Eliminating rRNA Aligned Reads: 102,029,713  </p>

<h2>6.  Convert unmapped reads (BAM file) back to fastQ format - DONE</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/convertToFastQ.brain.sh
</code></pre>

<h2>7.  Align non-ribosomal trimmed reads to strain specific genomes including mitochondrial chromosome - RUNNING</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/tophatAlign.brain.sh
</code></pre>

<p><strong><em>STOPPED HERE</em></strong></p>

<h2>8.  Sort and merge BAM files by strain</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/sortAndMerge.brain.sh
</code></pre>

<h2>9.  Strain-Specific Transcriptome Reconstruction (Ensembl-Guided)</h2>

<pre><code>qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/reconSHR.brain.sh
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/reconBNLx.brain.sh
</code></pre>

<h2>10.  Generate BigWig files</h2>

</body>

</html>

