BNLx and SHR Brain ribosomal RNA depleted total RNA
=========================
* 6 brain ribosomal RNA depleted total RNA samples, SHR1, SHR2, SHR3, BNLx1, BNLx2, and BNLx3
  * 2X100 paired end reads using the unstranded protocol

1. Unzip raw reads - DONE
-----------------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/unzip.yucca.sh
```

2. Consolidate reads by sample and determine number of reads sent for each sample - DONE
-------------------------------------------------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/countRawReads.sh
```

```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/NextGenSeq/BNLx.SHR.Brain.totalRNA/")
rawCounts = read.table(file="data/rawReadCounts.26Jun14.txt",sep=" ",header=FALSE,fill=TRUE)

rawCounts$readFrag = as.numeric(rawCounts$V3)
rawCounts$sample = unlist(lapply(strsplit(rawCounts$V1,split="/",fixed=TRUE),function(a) gsub(".R2.fq","",gsub(".R1.fq","",a[length(a)]))))

readFragments = aggregate(rawCounts$readFrag,by=list(sample=rawCounts$sample),sum)
readFragments$numPairedReads = prettyNum(readFragments$x/2,big.mark=",",scientific=FALSE)
readFragments$numReadFragments = prettyNum(readFragments$x,big.mark=",",scientific=FALSE)

readFragments=readFragments[,colnames(readFragments)!="x"]
forPrint = readFragments[,c("sample","numPairedReads","numReadFragments")]
colnames(forPrint) = c("sample","Number of Paired-End Reads","Number of Read Fragments")
```

Raw Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint,align=rep("c",ncol(readFragments)))
```

Total Number of Paired End Reads: `r prettyNum(sum(rawCounts$readFrag)/2,big.mark=",",scientific=FALSE)`  
Total Number of Read Fragments:  `r prettyNum(sum(rawCounts$readFrag),big.mark=",",scientific=FALSE)`  
Average Number of Paired End Reads Per Sample: `r prettyNum(sum(rawCounts$readFrag)/nrow(rawCounts),big.mark=",",scientific=FALSE)`  


3. Trim reads for adaptors and for quality - DONE
---------------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/trimReads.yucca.sh
```


4.  Characterize trimmed reads - DONE
-----------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/countTrimmedReads.sh
```
```{r,echo=FALSE}
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/NextGenSeq/BNLx.SHR.Brain.totalRNA/")

trimmed = read.table(file="data/trimmedReadCounts.26Jun14.txt",sep="",header=FALSE)
trimmed$file = unlist(lapply(strsplit(trimmed$V1,split="/",fixed=TRUE),function(a) a[length(a)]))
trimmed$sample = unlist(lapply(strsplit(trimmed$file,split=".",fixed=TRUE),function(a) a[1]))
trimmed$read = unlist(lapply(strsplit(trimmed$file,split=".",fixed=TRUE),function(a) unlist(lapply(strsplit(a[2],split="_",fixed=TRUE),function(a) a[1]))))

bySample = merge(trimmed[trimmed$read=="R1",c("sample","V2","V3")],trimmed[trimmed$read=="R2",c("sample","V2")],by="sample")
bySample$numReadFrag = bySample$V3*2
colnames(bySample) = c("sample","avgFragLength.R1","numReads","avgFragLength.R2","numReadFrag")

bySample = merge(readFragments,bySample,by="sample")
bySample$pctReadsAfterTrim = paste(sprintf("%.1f",round(100*bySample$numReads/as.numeric(gsub(",","",bySample$numPairedReads)),1)),"%",sep="")

forPrint2 = bySample[,c("sample","numPairedReads","numReadFragments","avgFragLength.R1","avgFragLength.R2","numReadFrag","pctReadsAfterTrim")]
forPrint2$avgFragLength.R1 = sprintf("%.1f",round(forPrint2$avgFragLength.R1,1))
forPrint2$avgFragLength.R2 = sprintf("%.1f",round(forPrint2$avgFragLength.R2,1))
forPrint2$numReadFrag = prettyNum(forPrint2$numReadFrag,big.mark=",")

colnames(forPrint2) = c("sample","Number of Paired-End Reads","Number of Read Fragments","Average Read Fragment Length After Trimming (first read fragment)","Average Read Fragment Length After Trimming (second read fragment)","Number of Read Fragments After Trimming","Percent of Read Fragments That Remained After Trimming")
```

Trimmed Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint2,align=rep("c",ncol(forPrint2)))
```

Total Number of Paired End Reads After Trimming: `r prettyNum(sum(bySample$numReads),big.mark=",",scientific=FALSE)`  
Total Number of Read Fragments After Trimming:  `r prettyNum(sum(bySample$numReadFrag),big.mark=",",scientific=FALSE)`  
Average Number of Paired End Reads Per Sample After Trimming: `r prettyNum(mean(bySample$numReads),big.mark=",",scientific=FALSE)`  

5.  Align trimmed reads to ribosomal RNA - DONE
-------------------------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/alignTo.rRNA.brain.01Jul14.sh
```

*tranfer files to phenogen server*  
get BNLx1.rRNA/align_summary.txt rRNA.summary.BNLx1.txt  
get BNLx2.rRNA/align_summary.txt rRNA.summary.BNLx2.txt  
get BNLx3.rRNA/align_summary.txt rRNA.summary.BNLx3.txt  
get SHR1.rRNA/align_summary.txt rRNA.summary.SHR1.txt  
get SHR2.rRNA/align_summary.txt rRNA.summary.SHR2.txt  
get SHR3.rRNA/align_summary.txt rRNA.summary.SHR3.txt  

```{r,echo=FALSE}
rm(list=ls())
setwd("/Volumes/LauraS/NextGenSeq/BNLx.SHR.Brain.totalRNA/data")
options(stringsAsFactors=FALSE)

sampleList = paste(rep(c("BNLx","SHR"),each=3),c(1:3),sep="")

rStats = c()
for(i in sampleList){
  x = read.table(file=paste("rRNA.summary.",i,".txt",sep=""),sep="\t",header=FALSE)
  sample = i
  numReads = as.numeric(gsub(":","",gsub("Input","",x[2,1])))       
  mappedLeft = as.numeric(gsub(":","",gsub("Mapped","",unlist(lapply(strsplit(x[3,1],split="(",fixed=TRUE),function(a) a[1])))))
  mappedRight = as.numeric(gsub(":","",gsub("Mapped ","",unlist(lapply(strsplit(x[7,1],split="(",fixed=TRUE),function(a) a[1])))))
  mappedPairs = as.numeric(gsub("Aligned pairs: ","",x[10,1]))  
  rStats = rbind(rStats,data.frame(sample,numReads,mappedLeft,mappedRight,mappedPairs))
}

rStats$numReadFrag = (rStats$mappedLeft + rStats$mappedRight)
rStats$numDeletedPairedReads = (rStats$mappedLeft + rStats$mappedRight - rStats$mappedPairs)

rStats = rStats[,c("sample","numReads","numReadFrag","mappedPairs","numDeletedPairedReads")]

rTable = data.frame(sample=rStats$sample,numTrimmedReads=prettyNum(rStats$numReads,big.mark=","),numDeleted=prettyNum(rStats$numDeletedPairedReads,big.mark=","),pctRiboReads=as.numeric(rStats$numDeletedPairedReads)/as.numeric(rStats$numReads),numRemaining=prettyNum(rStats$numReads-rStats$numDeletedPairedReads,big.mark=","))

rTable$pctRiboReads = paste(sprintf("%.1f",round(100*rTable$pctRiboReads,1)),"%",sep="")
colnames(rTable) = c("Sample","Number of Paired-End Reads After Trimming","Number of Paired-End Reads with at Least One Fragment Aligned to rRNA","Percent of Paired-End Reads Aligned to rRNA","Number of Paired-End Reads NOT Aligned to rRNA")
```

```{r, results='asis',echo=FALSE}
kable(rTable,align=rep("c",ncol(alignTable)))
```

Total Number of Paired End Reads After Eliminating rRNA Aligned Reads: `r prettyNum(sum(rStats$numReads-rStats$numDeletedPairedReads),big.mark=",",scientific=FALSE)`    
Average Number of Paired End Reads Per Sample After Eliminating rRNA Aligned Reads: `r prettyNum(mean(rStats$numReads-rStats$numDeletedPairedReads),big.mark=",",scientific=FALSE)`  

6.  Convert unmapped reads (BAM file) back to fastQ format - DONE
------------------------------------------------------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/convertToFastQ.brain.sh
```

7.  Align non-ribosomal trimmed reads to strain specific genomes including mitochondrial chromosome - DONE
----------------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/tophatAlign.brain.sh
```

8.  Sort and merge BAM files by strain -DONE
----------------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/sortAndMerge.brain.BNLx.sh
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/sortAndMerge.brain.SHR.sh
```

9. Characterize aligned reads - BNLx is DONE, SHR is RUNNING
-----------------------------
**move files**
```
cd /home/data/saba/BNLx.SHR.Brain.totalRNA/alignedReads
get ./BNLx1/align_summary.txt align_summary.BNLx1.txt
get ./BNLx2/align_summary.txt align_summary.BNLx2.txt
get ./BNLx3/align_summary.txt align_summary.BNLx3.txt
get ./SHR1/align_summary.txt align_summary.SHR1.txt
get ./SHR2/align_summary.txt align_summary.SHR2.txt
get ./SHR3/align_summary.txt align_summary.SHR3.txt
```

```{r,echo=FALSE,eval=TRUE}
sampleList=c("BNLx1","BNLx2","BNLx3","SHR1","SHR2","SHR3")

alignStats = c()
for(i in sampleList){
  x = read.table(file=paste("/Volumes/LauraS/NextGenSeq/BNLx.SHR.Brain.totalRNA/data/align_summary.",i,".txt",sep=""),sep="\t",header=FALSE)
  sample = i
  numReads = as.numeric(gsub(":","",gsub("Input","",x[2,1])))       
  mappedLeft = as.numeric(gsub(":","",gsub("Mapped","",unlist(lapply(strsplit(x[3,1],split="(",fixed=TRUE),function(a) a[1])))))
  mappedRight = as.numeric(gsub(":","",gsub("Mapped ","",unlist(lapply(strsplit(x[7,1],split="(",fixed=TRUE),function(a) a[1])))))
  multiLeft = as.numeric(gsub("of these: ","",unlist(lapply(strsplit(x[4,1],split="(",fixed=TRUE),function(a) a[1]))))
  multiRight = as.numeric(gsub("of these: ","",unlist(lapply(strsplit(x[8,1],split="(",fixed=TRUE),function(a) a[1]))))
  lotsLeft = as.numeric(gsub(" have >20)","",unlist(lapply(strsplit(x[4,1],split="(",fixed=TRUE),function(a) a[3]))))
  lotsRight = as.numeric(gsub(" have >20)","",unlist(lapply(strsplit(x[8,1],split="(",fixed=TRUE),function(a) a[3]))))
  mappedPairs = as.numeric(gsub("Aligned pairs: ","",x[10,1]))  
  multiPairs = as.numeric(gsub("of these: ","",unlist(lapply(strsplit(x[11,1],split="(",fixed=TRUE),function(a) a[1]))))
  discordPairs = as.numeric(gsub("and: ","",unlist(lapply(strsplit(x[12,1],split="(",fixed=TRUE),function(a) a[1]))))

  alignStats = rbind(alignStats,data.frame(sample,numReads,mappedLeft,mappedRight,multiLeft,multiRight,lotsLeft,lotsRight,mappedPairs,multiPairs,discordPairs))
}

alignTable = data.frame(t(alignStats))
colnames(alignTable) = alignTable[1,]
alignTable = alignTable[-1,]

alignTable$BNLx1.pct = as.numeric(alignTable$BNLx1)/as.numeric(alignTable["numReads","BNLx1"])
alignTable$BNLx2.pct = as.numeric(alignTable$BNLx2)/as.numeric(alignTable["numReads","BNLx2"])
alignTable$BNLx3.pct = as.numeric(alignTable$BNLx3)/as.numeric(alignTable["numReads","BNLx3"])
alignTable$SHR1.pct = as.numeric(alignTable$SHR1)/as.numeric(alignTable["numReads","SHR1"])
alignTable$SHR2.pct = as.numeric(alignTable$SHR2)/as.numeric(alignTable["numReads","SHR2"])
alignTable$SHR3.pct = as.numeric(alignTable$SHR3)/as.numeric(alignTable["numReads","SHR3"])

## format numbers
for(i in sampleList){
  alignTable[,i] = prettyNum(alignTable[,i],big.mark=",",scientific=FALSE)
  alignTable[,paste(i,".pct",sep="")] = paste(sprintf("%.2f", round(alignTable[,paste(i,".pct",sep="")]*100,2)),"%",sep="")
  }

## add labels
labels = read.table(file="/Volumes/LauraS/NextGenSeq/misc/alignLabels.txt",sep="\t",header=TRUE)
alignTable = merge(labels,alignTable,by.x="var",by.y=0)
alignTable = alignTable[order(alignTable$orderNum),]
alignTable = alignTable[,c("Label",paste(rep(sampleList,each=2),rep(c("",".pct"),length(sampleList)),sep=""))]
colnames(alignTable) = c("Label",paste(rep(sampleList,each=2),rep(c(" (num)"," (pct)"),length(sampleList)),sep=""))
```

### Statistics on Alignment to Genome
```{r, results='asis',echo=FALSE,eval=TRUE}
kable(alignTable,align=rep("c",ncol(alignTable)),row.names=FALSE)
```

Total Number of Aligned Paired End Reads: `r prettyNum(sum(alignStats$mappedPairs),big.mark=",",scientific=FALSE)`    
Total Number of Aligned Read Fragments: `r prettyNum(sum(alignStats[,c("mappedRight","mappedLeft")]),big.mark=",",scientific=FALSE)` 
Average Number of Aligned Read Fragments Per Sample: `r prettyNum(mean(rowSums(alignStats[,c("mappedRight","mappedLeft")])),big.mark=",",scientific=FALSE)`  

### Statistics By Chromosome and By Spike
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/countByChrom.brain.total.sh
```

```{r,echo=FALSE,eval=TRUE}
#rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("/Volumes/LauraS/NextGenSeq/BNLx.SHR.Brain.totalRNA/")

sampleList=c("BNLx1","BNLx2","BNLx3","SHR1","SHR2","SHR3")
for(sample in sampleList){
  x = read.table(file=paste("data/countByChr.",sample,".brain.txt",sep=""),header=FALSE,sep="\t")
  colnames(x) = c("id","length",sample,"V4")
  x = x[,-4]
  if(sample==sampleList[1]) chrCnts = x
  if(sample!=sampleList[1]) chrCnts = merge(chrCnts,x,by=c("id","length"))
}

chromo = chrCnts[grep("chr",chrCnts$id),]
pctBySampleChr = apply(chromo[,-c(1:2)],1,function(a) a/colSums(chromo[,-c(1:2)]))
colnames(pctBySampleChr) = chromo[,1]
pctBySampleChr = pctBySampleChr[,paste("chr",c(1:20,"X","M"),sep="")]
```

```{r,fig.width=12,fig.height=6,echo=FALSE}
boxplot(pctBySampleChr,ylab="percent of alignments",xlab="chromosome",main="Distribution Among Samples of Percent of Alignments By Chromosome",cex.axis=0.7)
```

```{r,fig.width=6,fig.height=6,echo=FALSE}
plot(hclust(as.dist(1-cor(t(pctBySampleChr))),method="average"),main="Comparison of Percent Alignments to Each Chromosome",ylab="1 minus correlation coefficient",sub="",xlab="")
```

10.  Strain-Specific Transcriptome Reconstruction (Ensembl-Guided) - BNLx and SHR are RUNNING
--------------------------
```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/reconBNLx.brain.sh
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/reconSHR.brain.sh
```

11.  Generate BigWig files - didn't run because of memory?
-------------------------

```
qsub -q smp /home/data/saba/BNLx.SHR.Brain.totalRNA/programs/createBigWig.brain.total.sh
```