---
title: "RI Panel - Batch 2 (Liver) Total RNA RNA-Seq"
author: "Laura Saba"
date: "July 21, 2015"
output:
  md_document:
    variant: markdown_github
---

RI Panel - Batch 2 (Liver) Total RNA RNA-Seq
========================================================

* 1 sample per strain (SHR,BXH2,HXB1,HXB3,HXB4,HXB5,HXB10,HXB15,HXB18,HXB20,HXB21,HXB22,HXB23,HXB24,HXB29,HXB31)
* tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic
* ribosomal RNA depleted total RNA
* 2X100 reads using the stranded protocol
* sequencing done by Beckman Coulter Genomics - received June 5, 2015
* synthetic spike-in were added to each sample 
  * ?? have mix 1; ?? have mix 2
  * 4 mL of 1:100 diluted spikes were added after RNA extraction but before rRNA-depletion
  * Information regarding the spike-ins can be found here: http://products.invitrogen.com/ivgn/product/4456739

1. Unzip Files - DONE
---------------
```
qsub -q smp -N liver.batch2 /storage/saba/Rat.RI.Liver.totalRNA.batch2/programs/unzip.yucca.sh
```

2. Determine number of reads sent for each sample - RUNNING
-------------------------------------------------
```
qsub -q smp -N cntRaw /storage/saba/Rat.RI.Liver.totalRNA.batch2/programs/countRawReads.liver.batch2.sh
```

```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
library(knitr)

base.folder = "~/Documents/NextGenSeq/Rat.RI.Liver.totalRNA.batch2/"

rawCounts = read.table(file=paste(base.folder,"data/rawReadCounts.22Jul15.txt",sep=""),sep=" ",header=FALSE,fill=TRUE)

rawCounts$readFrag = as.numeric(rawCounts$V3)
rawCounts$file = unlist(lapply(strsplit(rawCounts$V1,split="/",fixed=TRUE),function(a) a[length(a)]))
rawCounts$strain = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) a[1]))
rawCounts$sampleNum = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) gsub("liver","",a[2])))
rawCounts$sample = paste(rawCounts$strain,rawCounts$sampleNum,sep="_")

readFragments = aggregate(rawCounts$readFrag,by=list(sample=rawCounts$sample),sum)
readFragments$numPairedReads = prettyNum(readFragments$x,big.mark=",",scientific=FALSE)
readFragments$numReadFragments = prettyNum(readFragments$x*2,big.mark=",",scientific=FALSE)

readFragments=readFragments[,colnames(readFragments)!="x"]
forPrint = readFragments[,c("sample","numPairedReads","numReadFragments")]
colnames(forPrint) = c("sample","Number of Paired-End Reads","Number of Read Fragments")
```

Raw Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint,align=rep("c",ncol(readFragments)))
```

Total Number of Paired End Reads: `r prettyNum(sum(rawCounts$readFrag),big.mark=",",scientific=FALSE)`


Total Number of Read Fragments:  `r prettyNum(sum(rawCounts$readFrag*2),big.mark=",",scientific=FALSE)`  


Average Number of Paired End Reads Per Sample: `r prettyNum(sum(rawCounts$readFrag)/length(unique(rawCounts$sample)),big.mark=",",scientific=FALSE)`  

3. Trim reads for adaptors and for quality - DONE
---------------

```
qsub -q smp -N trimBatch2 /storage/saba/Rat.RI.Liver.totalRNA.batch2/programs/trimReads.liver.batch2.sh
```

4. Characterize Trimmed Reads - RUNNING
---------------
```
qsub -q smp -N cntTrimmed2 /storage/saba/Rat.RI.Liver.totalRNA.batch2/programs/countTrimmedReads.liver.batch2.sh
```
```{r,echo=FALSE}
options(stringsAsFactors=FALSE)
setwd("~/Documents/NextGenSeq/Rat.RI.Liver.totalRNA.batch2/")

trimmed = read.table(file="data/trimmedReadCounts.15Jul15.txt",sep="",header=FALSE)
trimmed$file = unlist(lapply(strsplit(trimmed$V1,split="/",fixed=TRUE),function(a) a[length(a)]))
trimmed$sample = unlist(lapply(strsplit(trimmed$file,split="_",fixed=TRUE),function(a) gsub("liver","",paste(a[1],a[2],sep="_"))))
trimmed$read = unlist(lapply(strsplit(trimmed$file,split="_",fixed=TRUE),function(a) a[5]))
trimmed$lane = unlist(lapply(strsplit(trimmed$file,split="_",fixed=TRUE),function(a) a[4]))

bySample = merge(trimmed[trimmed$read=="R1",c("sample","lane","V2","V3")],trimmed[trimmed$read=="R2",c("sample","lane","V2")],by=c("sample","lane"))
bySample$numReadFrag = bySample$V3*2
colnames(bySample) = c("sample","lane","avgFragLength.R1","numReads","avgFragLength.R2","numReadFrag")

rawCounts$lane = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) a[4])) 
rawCounts = rawCounts[grep("_R1",rawCounts$file),]

bySample = merge(rawCounts,bySample,by=c("sample","lane"))
bySample$pctReadsAfterTrim = paste(sprintf("%.1f",round(100*bySample$numReads/bySample$readFrag,1)),"%",sep="")

bySample$numPairedReads = bySample$readFrag
bySample$numReadFragments = bySample$readFrag*2

forPrint2 = bySample[,c("sample","lane","numPairedReads","numReadFragments","avgFragLength.R1","avgFragLength.R2","numReadFrag","pctReadsAfterTrim")]
forPrint2$avgFragLength.R1 = sprintf("%.1f",round(forPrint2$avgFragLength.R1,1))
forPrint2$avgFragLength.R2 = sprintf("%.1f",round(forPrint2$avgFragLength.R2,1))
forPrint2$numReadFrag = prettyNum(forPrint2$numReadFrag,big.mark=",",scientific=FALSE)
forPrint2$numPairedReads = prettyNum(forPrint2$numPairedReads,big.mark=",",scientific=FALSE)
forPrint2$numReadFragments = prettyNum(forPrint2$numReadFragments,big.mark=",",scientific=FALSE)


colnames(forPrint2) = c("sample","lane","Number of Paired-End Reads","Number of Read Fragments","Average Read Fragment Length After Trimming (first read fragment)","Average Read Fragment Length After Trimming (second read fragment)","Number of Read Fragments After Trimming","Percent of Read Fragments That Remained After Trimming")
```

Trimmed Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint2,align=rep("c",ncol(forPrint2)))
```

Total Number of Paired End Reads After Trimming: `r prettyNum(sum(as.numeric(bySample$numReads)),big.mark=",",scientific=FALSE)`   


Total Number of Read Fragments After Trimming:  `r prettyNum(sum(bySample$numReadFrag),big.mark=",",scientific=FALSE)`  


Average Number of Paired End Reads Per Sample After Trimming: `r prettyNum(mean(bySample$numReads),big.mark=",",scientific=FALSE)`  


5. Align to ribosomal RNA - 
---------------------------
```
qsub -q smp -N rRNA2 /storage/saba/Rat.RI.Liver.totalRNA.batch2/programs/alignTo.rRNA.liver.totalRNA.batch2.sh
```


**move files**
```{r,echo=FALSE,eval=FALSE}
## NOT R code, but don't want in Md file
get BXH2_1.rRNA/align_summary.txt rRNA.summary.BXH2_1.txt
get HXB1_1.rRNA/align_summary.txt rRNA.summary.HXB1_1.txt
get HXB3_1.rRNA/align_summary.txt rRNA.summary.HXB3_1.txt
get HXB4_1.rRNA/align_summary.txt rRNA.summary.HXB4_1.txt
get HXB5_1.rRNA/align_summary.txt rRNA.summary.HXB5_1.txt
get HXB10_1.rRNA/align_summary.txt rRNA.summary.HXB10_1.txt
get HXB15_1.rRNA/align_summary.txt rRNA.summary.HXB15_1.txt
get HXB18_1.rRNA/align_summary.txt rRNA.summary.HXB18_1.txt
get HXB20_1.rRNA/align_summary.txt rRNA.summary.HXB20_1.txt
get HXB21_1.rRNA/align_summary.txt rRNA.summary.HXB21_1.txt
get HXB22_1.rRNA/align_summary.txt rRNA.summary.HXB22_1.txt
get HXB23_1.rRNA/align_summary.txt rRNA.summary.HXB23_1.txt
get HXB24_1.rRNA/align_summary.txt rRNA.summary.HXB24_1.txt
get HXB29_1.rRNA/align_summary.txt rRNA.summary.HXB29_1.txt
get HXB31_1.rRNA/align_summary.txt rRNA.summary.HXB31_1.txt
get SHR_1.rRNA/align_summary.txt rRNA.summary.SHR_1.txt
```

Proportion of Reads Aligned to rRNA
-----------------------------------
This batch was subjected to 2 washes with riboZero.  

```{r,echo=FALSE,eval=TRUE}
#rm(list=ls())
data.folder="~/Documents/NextGenSeq/Rat.RI.Liver.totalRNA.batch2/data"
options(stringsAsFactors=FALSE)

sampleList = rawCounts$sample

rStats = c()
for(i in sampleList){
  x = read.table(file=paste(data.folder,"/rRNA.summary.",i,".txt",sep=""),sep="\t",header=FALSE)
  sample = i
  numReads = as.numeric(gsub(":","",gsub("Input","",x[2,1])))       
  mappedLeft = as.numeric(gsub(":","",gsub("Mapped","",unlist(lapply(strsplit(x[3,1],split="(",fixed=TRUE),function(a) a[1])))))
  mappedRight = as.numeric(gsub(":","",gsub("Mapped","",unlist(lapply(strsplit(x[7,1],split="(",fixed=TRUE),function(a) a[1])))))
  mappedPairs = as.numeric(gsub("Aligned pairs: ","",x[10,1]))  
  rStats = rbind(rStats,data.frame(sample,numReads,mappedLeft,mappedRight,mappedPairs))
}

rStats$numReadFrag = (rStats$mappedLeft + rStats$mappedRight)
rStats$numDeletedPairedReads = (rStats$mappedLeft + rStats$mappedRight - rStats$mappedPairs)

rStats = rStats[,c("sample","numReads","numReadFrag","mappedPairs","numDeletedPairedReads")]

rTable = data.frame(sample=rStats$sample,numTrimmedReads=prettyNum(rStats$numReads,big.mark=",",scientific=FALSE),numDeleted=prettyNum(rStats$numDeletedPairedReads,big.mark=",",scientific=FALSE),pctRiboReads=as.numeric(rStats$numDeletedPairedReads)/as.numeric(rStats$numReads),numRemaining=prettyNum(rStats$numReads-rStats$numDeletedPairedReads,big.mark=",",scientific=FALSE))

rTable$pctRiboReads = paste(sprintf("%.1f",round(100*rTable$pctRiboReads,1)),"%",sep="")
colnames(rTable) = c("Sample","Number of Paired-End Reads After Trimming","Number of Paired-End Reads with at Least One Fragment Aligned to rRNA","Percent of Paired-End Reads Aligned to rRNA","Number of Paired-End Reads NOT Aligned to rRNA")
```

```{r, results='asis',echo=FALSE}
kable(rTable,align=rep("c",ncol(rTable)))
```

6. Convert to fastq files
---------------------------
```
/hi-seq/sharedPrograms/bamToFastQ_py.phenogen.sh /hi-seq/RI.Brain.totalRNA.09Jun15/alignedReads/BNLx_3.rRNA &
/hi-seq/sharedPrograms/bamToFastQ_py.phenogen.sh /hi-seq/RI.Brain.totalRNA.09Jun15/alignedReads/SHR_1.rRNA &
/hi-seq/sharedPrograms/bamToFastQ_py.phenogen.sh /hi-seq/RI.Brain.totalRNA.09Jun15/alignedReads/SHR_3.rRNA &

