RI Panel - Batch 2 (Brain) Total RNA RNA-Seq
========================================================

* 1 sample per strain (SHR, BXH2, HXB10, HXB15, HXB18, HXB1, HXB20, HXB21, HXB22, HXB23, HXB24, HXB29, HXB31, HXB3, HXB4, HXB5, SHR)
* tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic
* ribosomal RNA depleted total RNA
* 2X100 reads using the stranded protocol
* sequencing done by Beckman Coulter Genomics - received September 24, 2014
  * labeled UColorado Tabakoff Batch 5 by Beckman
* synthetic spike-in were added to each sample 
  * HXB1_1, HXB3_1, HXB4_1, HXB15_1, HXB23_1, HXB24_1, HXB31_1, and SHR_1 have mix 1
  * HXB5_1, HXB10_1, HXB18_1, HXB20_1, HXB21_1, HXB22_1, HXB29_1, and BXH2_1 have mix 2
  * 4 mL of 1:100 diluted spikes were added after RNA extraction but before rRNA-depletion
  * Information regarding the spike-ins can be found here: http://products.invitrogen.com/ivgn/product/4456739

1. Unzip Files - DONE
---------------
```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/unzip.yucca.sh
```

2. Determine number of reads sent for each sample - DONE
-------------------------------------------------
```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/countRawReads.brain.batch2.sh
```

```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("~/Documents/NextGenSeq/Rat.RI.Brain.totalRNA.batch2/")
rawCounts = read.table(file="data/rawReadCounts.25Sep14.txt",sep=" ",header=FALSE,fill=TRUE)

rawCounts$readFrag = as.numeric(rawCounts$V3)
rawCounts$file = unlist(lapply(strsplit(rawCounts$V1,split="/",fixed=TRUE),function(a) a[length(a)]))
rawCounts$strain = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) a[1]))
rawCounts$sampleNum = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) gsub("brain","",a[2])))
rawCounts$sample = paste(rawCounts$strain,rawCounts$sampleNum,sep="_")

readFragments = aggregate(rawCounts$readFrag,by=list(sample=rawCounts$sample),sum)
readFragments$numPairedReads = prettyNum(readFragments$x/2,big.mark=",",scientific=FALSE)
readFragments$numReadFragments = prettyNum(readFragments$x,big.mark=",",scientific=FALSE)

readFragments=readFragments[,colnames(readFragments)!="x"]
forPrint = readFragments[,c("sample","numPairedReads","numReadFragments")]
colnames(forPrint) = c("sample","Number of Paired-End Reads","Number of Read Fragments")
```

Raw Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint,align=rep("c",ncol(readFragments)))
```

Total Number of Paired End Reads: `r prettyNum(sum(rawCounts$readFrag)/2,big.mark=",",scientific=FALSE)`  
Total Number of Read Fragments:  `r prettyNum(sum(rawCounts$readFrag),big.mark=",",scientific=FALSE)`  
Average Number of Paired End Reads Per Sample: `r prettyNum(sum(rawCounts$readFrag/2)/length(unique(rawCounts$sample)),big.mark=",",scientific=FALSE)`  

3. Trim reads for adaptors and for quality - RUNNING
---------------

```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/trimReads.brain.batch2.sh
```