<!DOCTYPE html>
<!-- saved from url=(0014)about:internet -->
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=9" >

<title>RI Panel - Batch 2 (Brain) Total RNA RNA-Seq</title>

<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 12px;
   margin: 8px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 { 
   font-size:2.2em; 
}

h2 { 
   font-size:1.8em; 
}

h3 { 
   font-size:1.4em; 
}

h4 { 
   font-size:1.0em; 
}

h5 { 
   font-size:0.9em; 
}

h6 { 
   font-size:0.8em; 
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre {	
   margin-top: 0;
   max-width: 95%;
   border: 1px solid #ccc;
   white-space: pre-wrap;
}

pre code {
   display: block; padding: 0.5em;
}

code.r, code.cpp {
   background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * { 
      background: transparent !important; 
      color: black !important; 
      filter:none !important; 
      -ms-filter: none !important; 
   }

   body { 
      font-size:12pt; 
      max-width:100%; 
   }
       
   a, a:visited { 
      text-decoration: underline; 
   }

   hr { 
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote { 
      padding-right: 1em; 
      page-break-inside: avoid; 
   }

   tr, img { 
      page-break-inside: avoid; 
   }

   img { 
      max-width: 100% !important; 
   }

   @page :left { 
      margin: 15mm 20mm 15mm 10mm; 
   }
     
   @page :right { 
      margin: 15mm 10mm 15mm 20mm; 
   }

   p, h2, h3 { 
      orphans: 3; widows: 3; 
   }

   h2, h3 { 
      page-break-after: avoid; 
   }
}

</style>





</head>

<body>
<h1>RI Panel - Batch 2 (Brain) Total RNA RNA-Seq</h1>

<ul>
<li>1 sample per strain (SHR, BXH2, HXB10, HXB15, HXB18, HXB1, HXB20, HXB21, HXB22, HXB23, HXB24, HXB29, HXB31, HXB3, HXB4, HXB5, SHR)</li>
<li>tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic</li>
<li>ribosomal RNA depleted total RNA</li>
<li>2X100 reads using the stranded protocol</li>
<li>sequencing done by Beckman Coulter Genomics - received September 24, 2014

<ul>
<li>labeled UColorado Tabakoff Batch 5 by Beckman</li>
</ul></li>
<li>synthetic spike-in were added to each sample 

<ul>
<li>HXB1_1, HXB3_1, HXB4_1, HXB15_1, HXB23_1, HXB24_1, HXB31_1, and SHR_1 have mix 1</li>
<li>HXB5_1, HXB10_1, HXB18_1, HXB20_1, HXB21_1, HXB22_1, HXB29_1, and BXH2_1 have mix 2</li>
<li>4 mL of 1:100 diluted spikes were added after RNA extraction but before rRNA-depletion</li>
<li>Information regarding the spike-ins can be found here: <a href="http://products.invitrogen.com/ivgn/product/4456739">http://products.invitrogen.com/ivgn/product/4456739</a></li>
</ul></li>
</ul>

<h2>1. Unzip Files - DONE</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/unzip.yucca.sh
</code></pre>

<h2>2. Determine number of reads sent for each sample - DONE</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/countRawReads.brain.batch2.sh
</code></pre>

<h2>Raw Reads/Read Fragments</h2>

<table><thead>
<tr>
<th align="center">sample</th>
<th align="center">Number of Paired-End Reads</th>
<th align="center">Number of Read Fragments</th>
</tr>
</thead><tbody>
<tr>
<td align="center">BXH2_1</td>
<td align="center">174,413,815</td>
<td align="center">348,827,630</td>
</tr>
<tr>
<td align="center">HXB1_1</td>
<td align="center">159,129,032</td>
<td align="center">318,258,064</td>
</tr>
<tr>
<td align="center">HXB10_1</td>
<td align="center">194,539,473</td>
<td align="center">389,078,946</td>
</tr>
<tr>
<td align="center">HXB15_1</td>
<td align="center">177,057,675</td>
<td align="center">354,115,350</td>
</tr>
<tr>
<td align="center">HXB18_1</td>
<td align="center">168,482,405</td>
<td align="center">336,964,810</td>
</tr>
<tr>
<td align="center">HXB20_1</td>
<td align="center">147,520,151</td>
<td align="center">295,040,302</td>
</tr>
<tr>
<td align="center">HXB21_1</td>
<td align="center">163,232,081</td>
<td align="center">326,464,162</td>
</tr>
<tr>
<td align="center">HXB22_1</td>
<td align="center">134,629,467</td>
<td align="center">269,258,934</td>
</tr>
<tr>
<td align="center">HXB23_1</td>
<td align="center">166,909,502</td>
<td align="center">333,819,004</td>
</tr>
<tr>
<td align="center">HXB24_1</td>
<td align="center">114,577,224</td>
<td align="center">229,154,448</td>
</tr>
<tr>
<td align="center">HXB29_1</td>
<td align="center">177,121,877</td>
<td align="center">354,243,754</td>
</tr>
<tr>
<td align="center">HXB3_1</td>
<td align="center">204,742,542</td>
<td align="center">409,485,084</td>
</tr>
<tr>
<td align="center">HXB31_1</td>
<td align="center">143,642,905</td>
<td align="center">287,285,810</td>
</tr>
<tr>
<td align="center">HXB4_1</td>
<td align="center">192,423,216</td>
<td align="center">384,846,432</td>
</tr>
<tr>
<td align="center">HXB5_1</td>
<td align="center">223,080,147</td>
<td align="center">446,160,294</td>
</tr>
<tr>
<td align="center">SHR_1</td>
<td align="center">118,513,047</td>
<td align="center">237,026,094</td>
</tr>
</tbody></table>

<p>Total Number of Paired End Reads: 2,660,014,559<br/>
Total Number of Read Fragments:  5,320,029,118<br/>
Average Number of Paired End Reads Per Sample: 166,250,910  </p>

<h2>3. Trim reads for adaptors and for quality - RUNNING</h2>

<pre><code>qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/trimReads.brain.batch2.sh
</code></pre>

</body>

</html>

