RI Panel - Batch 2 (Brain) Total RNA RNA-Seq
========================================================

* 1 sample per strain (SHR, BXH2, HXB10, HXB15, HXB18, HXB1, HXB20, HXB21, HXB22, HXB23, HXB24, HXB29, HXB31, HXB3, HXB4, HXB5, SHR)
* tissues received from Dr. Michal Pravenec at Academy of Sciences of the Czech Republic, Prague, Czech Republic
* ribosomal RNA depleted total RNA
* 2X100 reads using the stranded protocol
* sequencing done by Beckman Coulter Genomics - received September 24, 2014
  * labeled UColorado Tabakoff Batch 5 by Beckman
* synthetic spike-in were added to each sample 
  * HXB1_1, HXB3_1, HXB4_1, HXB15_1, HXB23_1, HXB24_1, HXB31_1, and SHR_1 have mix 1
  * HXB5_1, HXB10_1, HXB18_1, HXB20_1, HXB21_1, HXB22_1, HXB29_1, and BXH2_1 have mix 2
  * 4 mL of 1:100 diluted spikes were added after RNA extraction but before rRNA-depletion
  * Information regarding the spike-ins can be found here: http://products.invitrogen.com/ivgn/product/4456739

1. Unzip Files - DONE
---------------
```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/unzip.yucca.sh
```

2. Determine number of reads sent for each sample - DONE
-------------------------------------------------
```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/countRawReads.brain.batch2.sh
```

```{r,echo=FALSE}
rm(list=ls())
options(stringsAsFactors=FALSE)
setwd("~/Documents/NextGenSeq/Rat.RI.Brain.totalRNA.batch2/")
rawCounts = read.table(file="data/rawReadCounts.25Sep14.txt",sep=" ",header=FALSE,fill=TRUE)

rawCounts$readFrag = as.numeric(rawCounts$V3)
rawCounts$file = unlist(lapply(strsplit(rawCounts$V1,split="/",fixed=TRUE),function(a) a[length(a)]))
rawCounts$strain = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) a[1]))
rawCounts$sampleNum = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) gsub("brain","",a[2])))
rawCounts$sample = paste(rawCounts$strain,rawCounts$sampleNum,sep="_")

readFragments = aggregate(rawCounts$readFrag,by=list(sample=rawCounts$sample),sum)
readFragments$numPairedReads = prettyNum(readFragments$x/2,big.mark=",",scientific=FALSE)
readFragments$numReadFragments = prettyNum(readFragments$x,big.mark=",",scientific=FALSE)

readFragments=readFragments[,colnames(readFragments)!="x"]
forPrint = readFragments[,c("sample","numPairedReads","numReadFragments")]
colnames(forPrint) = c("sample","Number of Paired-End Reads","Number of Read Fragments")
```

Raw Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint,align=rep("c",ncol(readFragments)))
```

Total Number of Paired End Reads: `r prettyNum(sum(rawCounts$readFrag)/2,big.mark=",",scientific=FALSE)`  
Total Number of Read Fragments:  `r prettyNum(sum(rawCounts$readFrag),big.mark=",",scientific=FALSE)`  
Average Number of Paired End Reads Per Sample: `r prettyNum(sum(rawCounts$readFrag/2)/length(unique(rawCounts$sample)),big.mark=",",scientific=FALSE)`  

3. Trim reads for adaptors and for quality - DONE
---------------

```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/trimReads.brain.batch2.sh
```

4. Characterize Trimmed Reads - DONE
---------------
```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/countTrimmedReads.sh
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/compressRawReads.batch2.sh
```

```{r,echo=FALSE}
options(stringsAsFactors=FALSE)
setwd("~/Documents/NextGenSeq/Rat.RI.Brain.totalRNA.batch2/")

trimmed = read.table(file="data/trimmedReadCounts.10Jan15.txt",sep="",header=FALSE)
trimmed$file = unlist(lapply(strsplit(trimmed$V1,split="/",fixed=TRUE),function(a) a[6]))
trimmed$sample = unlist(lapply(strsplit(trimmed$file,split="_",fixed=TRUE),function(a) gsub("brain","",paste(a[1],a[2],sep="_"))))
trimmed$read = unlist(lapply(strsplit(trimmed$file,split="_",fixed=TRUE),function(a) a[5]))
trimmed$lane = unlist(lapply(strsplit(trimmed$file,split="_",fixed=TRUE),function(a) a[4]))

bySample = merge(trimmed[trimmed$read=="R1",c("sample","lane","V2","V3")],trimmed[trimmed$read=="R2",c("sample","lane","V2")],by=c("sample","lane"))
bySample$numReadFrag = bySample$V3*2
colnames(bySample) = c("sample","lane","avgFragLength.R1","numReads","avgFragLength.R2","numReadFrag")

rawCounts$lane = unlist(lapply(strsplit(rawCounts$file,split="_",fixed=TRUE),function(a) a[4]))
rawCounts = rawCounts[grep("_R1",rawCounts$file),]

bySample = merge(rawCounts,bySample,by=c("sample","lane"))
bySample$pctReadsAfterTrim = paste(sprintf("%.1f",round(100*bySample$numReads/bySample$readFrag,1)),"%",sep="")

bySample$numPairedReads = bySample$readFrag
bySample$numReadFragments = bySample$readFrag*2

forPrint2 = bySample[,c("sample","lane","numPairedReads","numReadFragments","avgFragLength.R1","avgFragLength.R2","numReadFrag","pctReadsAfterTrim")]
forPrint2$avgFragLength.R1 = sprintf("%.1f",round(forPrint2$avgFragLength.R1,1))
forPrint2$avgFragLength.R2 = sprintf("%.1f",round(forPrint2$avgFragLength.R2,1))
forPrint2$numReadFrag = prettyNum(forPrint2$numReadFrag,big.mark=",",scientific=FALSE)
forPrint2$numPairedReads = prettyNum(forPrint2$numPairedReads,big.mark=",",scientific=FALSE)
forPrint2$numReadFragments = prettyNum(forPrint2$numReadFragments,big.mark=",",scientific=FALSE)

colnames(forPrint2) = c("sample","lane","Number of Paired-End Reads","Number of Read Fragments","Average Read Fragment Length After Trimming (first read fragment)","Average Read Fragment Length After Trimming (second read fragment)","Number of Read Fragments After Trimming","Percent of Read Fragments That Remained After Trimming")
```

Trimmed Reads/Read Fragments
---------------------------

```{r, results='asis',echo=FALSE}
kable(forPrint2,align=rep("c",ncol(forPrint2)))
```

Total Number of Paired End Reads After Trimming: `r prettyNum(sum(as.numeric(bySample$numReads)),big.mark=",",scientific=FALSE)`  
Total Number of Read Fragments After Trimming:  `r prettyNum(sum(bySample$numReadFrag),big.mark=",",scientific=FALSE)`  
Average Number of Paired End Reads Per Sample After Trimming: `r prettyNum(2*mean(bySample$numReads),big.mark=",",scientific=FALSE)`  

5. Align to ribosomal RNA - RUNNING
---------------------------
```
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/alignTo.rRNA.brain.totalRNA.batch2.sh
qsub -q smp /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/restart.rRNAalign.batch2.sh

qsub -q compute -N HXB5_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB5_1brain_TGACCA /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB21_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB21_1brain_CTTGTA /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB22_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB22_1brain_GTGAAA /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB23_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB23_1brain_GTCCGC /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB24_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB24_1brain_GTCCGC /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB29_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB29_1brain_CAGATC /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB31_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB31_1brain_ATGTCA /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB3_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB3_1brain_GCCAAT /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB4_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB4_1brain_ACAGTG /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N SHR_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ SHR_1brain_CCGTCC /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB15_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB15_1brain_AGTTCC /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB18_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB18_1brain_AGTCAA /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT
qsub -q compute -N HXB20_1 -l nodes=1:ppn=12 -V -- /storage/saba/Rat.RI.Brain.totalRNA.batch2/programs/tophat.rRNA.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/rawReads/ HXB20_1brain_AGTTCC /storage/saba/index/rmsk.rRNA.RGSC5.0.rn5 AGATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT

```

**move files**
```{r,echo=FALSE,eval=FALSE}
## NOT R code, but don't want in Md file
get BXH2_1.rRNA/align_summary.txt rRNA.summary.BXH2_1.txt 
get HXB15_1.rRNA/align_summary.txt rRNA.summary.HXB15_1.txt  
get HXB21_1.rRNA/align_summary.txt rRNA.summary.HXB21_1.txt 
get HXB24_1.rRNA/align_summary.txt rRNA.summary.HXB24_1.txt  
get HXB3_1.rRNA/align_summary.txt rRNA.summary.HXB3_1.txt  
get SHR_1.rRNA/align_summary.txt rRNA.summary.SHR_1.txt
get HXB10_1.rRNA/align_summary.txt rRNA.summary.HXB10_1.txt 
get HXB18_1.rRNA/align_summary.txt rRNA.summary.HXB18_1.txt  
get HXB22_1.rRNA/align_summary.txt rRNA.summary.HXB22_1.txt  
get HXB29_1.rRNA/align_summary.txt rRNA.summary.HXB29_1.txt  
get HXB4_1.rRNA/align_summary.txt rRNA.summary.HXB4_1.txt
get HXB1_1.rRNA/align_summary.txt rRNA.summary.HXB1_1.txt   
get HXB20_1.rRNA/align_summary.txt rRNA.summary.HXB20_1.txt 
get HXB23_1.rRNA/align_summary.txt rRNA.summary.HXB23_1.txt 
get HXB5_1.rRNA/align_summary.txt rRNA.summary.HXB5_1.txt
get HXB31_1.rRNA/align_summary.txt rRNA.summary.HXB31_1.txt 


```

Proportion of Reads Aligned to rRNA
-----------------------------------
This batch was only subjected to 1 wash with riboZero.  The next batch received two washes.  
```{r,echo=FALSE,eval=TRUE}
#rm(list=ls())
setwd("~/Documents/NextGenSeq/Rat.RI.Brain.totalRNA.batch2/data")
options(stringsAsFactors=FALSE)

sampleList = unique(forPrint2$sample)

rStats = c()
for(i in sampleList){
  x = read.table(file=paste("rRNA.summary.",i,".txt",sep=""),sep="\t",header=FALSE)
  sample = i
  numReads = as.numeric(gsub(":","",gsub("Input","",x[2,1])))       
  mappedLeft = as.numeric(gsub(":","",gsub("Mapped","",unlist(lapply(strsplit(x[3,1],split="(",fixed=TRUE),function(a) a[1])))))
  mappedRight = as.numeric(gsub(":","",gsub("Mapped ","",unlist(lapply(strsplit(x[7,1],split="(",fixed=TRUE),function(a) a[1])))))
  mappedPairs = as.numeric(gsub("Aligned pairs: ","",x[10,1]))  
  rStats = rbind(rStats,data.frame(sample,numReads,mappedLeft,mappedRight,mappedPairs))
}

rStats$numReadFrag = (rStats$mappedLeft + rStats$mappedRight)
rStats$numDeletedPairedReads = (rStats$mappedLeft + rStats$mappedRight - rStats$mappedPairs)

rStats = rStats[,c("sample","numReads","numReadFrag","mappedPairs","numDeletedPairedReads")]

rTable = data.frame(sample=rStats$sample,numTrimmedReads=prettyNum(rStats$numReads,big.mark=",",scientific=FALSE),numDeleted=prettyNum(rStats$numDeletedPairedReads,big.mark=",",scientific=FALSE),pctRiboReads=as.numeric(rStats$numDeletedPairedReads)/as.numeric(rStats$numReads),numRemaining=prettyNum(rStats$numReads-rStats$numDeletedPairedReads,big.mark=",",scientific=FALSE))

rTable$pctRiboReads = paste(sprintf("%.1f",round(100*rTable$pctRiboReads,1)),"%",sep="")
colnames(rTable) = c("Sample","Number of Paired-End Reads After Trimming","Number of Paired-End Reads with at Least One Fragment Aligned to rRNA","Percent of Paired-End Reads Aligned to rRNA","Number of Paired-End Reads NOT Aligned to rRNA")
```

```{r, results='asis',echo=FALSE}
kable(rTable,align=rep("c",ncol(rTable)))
```

6.  Convert unmapped.bam file into fastq files - 
-----------------------------------------------
```
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/BXH2_1.rRNA
wc -l /hi-seq/RI.Brain.totalRNA.24Sep14/cleanedReads/BXH2_1.no_rRNA.*.fq > /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/BXH2_1.rRNA/numUnmappedFQ.txt

/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB10_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB1_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB15_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB18_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB20_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB21_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB22_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB23_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB24_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB29_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB31_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB3_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB4_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/HXB5_1.rRNA
/data2/saba/sharedPrograms/bamToFastQ.buildup.phenogen.sh /hi-seq/RI.Brain.totalRNA.24Sep14/alignedReads/SHR_1.rRNA

qsub -q compute -N HXB31_1 -l nodes=1:ppn=12 -V -- /storage/saba/sharedPrograms/bamToFastQ_py.sh /storage/saba/Rat.RI.Brain.totalRNA.batch2/alignedReads/HXB31_1.rRNA
get unmapped.end1.fq HXB31_1.no_rRNA.end1.fq
get unmapped.end2.fq HXB31_1.no_rRNA.end2.fq



```
```{r,echo=FALSE,eval=FALSE}
## NOT R code, but don't want in Md file
get BXH2_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.BXH2_1.txt
get HXB1_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB1_1.txt
get HXB10_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB10_1.txt 
get HXB15_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB15_1.txt 
get HXB18_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB18_1.txt
get HXB20_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB20_1.txt 
get HXB21_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB21_1.txt 
get HXB22_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB22_1.txt
get HXB23_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB23_1.txt 
get HXB24_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB24_1.txt 
get HXB29_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB29_1.txt 
get HXB3_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB3_1.txt 
get HXB4_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB4_1.txt
get HXB5_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB5_1.txt
get SHR_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.SHR_1.txt
get HXB31_1.rRNA/numUnmappedFQ.txt unmappedFQ.summary.HXB31_1.txt
```

```{r,echo=FALSE,eval=TRUE}
setwd("~/Documents/NextGenSeq/Rat.RI.Brain.totalRNA.batch2/data")
options(stringsAsFactors=FALSE)

unmappedStats = c()
for(i in sampleList){
  x = read.table(file=paste("unmappedFQ.summary.",i,".txt",sep=""),sep=" ",header=FALSE,fill=TRUE)
  sample = i
  numReads = as.numeric(x[1,3])/4
  unmappedStats = rbind(unmappedStats,data.frame(sample,numReads))
}

rownames(unmappedStats) = unmappedStats$sample

numCorrect = sum(as.numeric(gsub(",","",rTable[,"Number of Paired-End Reads NOT Aligned to rRNA"],fixed=TRUE))==unmappedStats[rTable$Sample,2])

numSamples = nrow(unmappedStats)
```

`r numCorrect` out of `r numSamples` of the fastq files were created correctly.

7.  Quantitate using RSEM and Ensembl Transcripts - RUNNING
------------------------------------------------------
```
export PATH=/usr/local/rsem/bin:$PATH

rsem-prepare-reference --bowtie2 --gtf /data2/saba/NextGenSeq/source/ensemblGenes.RGSC5.0.rn5.CLEANED.gtf /data2/saba/index/rn5_wSpikes.fa /data2/saba/index/RSEM/rn5.ensembl.wSpikes &

##  DONE  ##
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh BXH2_1 646 &
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB10_1 6461 &
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB1_1 6462 &
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB15_1 6463
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB18_1 6464
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB20_1 6465
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB21_1 6466
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB22_1 6467
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB23_1 6468
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB24_1 6469
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB29_1 6470
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB31_1 6471
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB3_1 6472
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB4_1 6473
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh HXB5_1 6474
/data2/saba/BxH.HxB.Rats/RNA-Seq/programs/quantRSEM.batch2.sh SHR_1 6575
```

```{r}
options(stringsAsFactors=FALSE)
count.path = "/Volumes/saba/BxH.HxB.Rats/RNA-Seq/quantitation/rsem.Ensembl/"

sampleList = c("BXH2_1","HXB10_1","HXB1_1","HXB15_1","HXB18_1")

for(i in sampleList){
  y = read.table(file=paste(count.path,i,".b2.isoforms.results",sep=""),sep="\t",header=TRUE)
  x = data.frame(sample=i,spikeCnt=sum(y$expected_count[grep("ERCC",y$transcript_id)]),totCnt=sum(y$expected_count))
  if(i==sampleList[1]) spikeSummary=x
  if(i!=sampleList[1]) spikeSummary=rbind(spikeSummary,x)
  }

spikeSummary$pct = paste(sprintf("%.1f",round(spikeSummary$spikeCnt/spikeSummary$totCnt,2)),"%",sep="")

spikeSummary$spikeCnt = prettyNum(spikeSummary$spikeCnt,big.mark=",",scientific=FALSE)
spikeSummary$totCnt = prettyNum(spikeSummary$totCnt,big.mark=",",scientific=FALSE)

colnames(spikeSummary) = c("Sample","Expect Read Count - Spikes","Expect Read Count - Total","Percent of Total Reads Aligned to Synthetic Spike-Ins")
```

